{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Copyright (C) 2019-2021  Martin Engqvist\n",
    "\n",
    "This program is free software: you can redistribute it and/or modify\n",
    "it under the terms of the GNU General Public License as published by\n",
    "the Free Software Foundation, either version 3 of the License, or\n",
    "(at your option) any later version.\n",
    "This program is distributed in the hope that it will be useful,\n",
    "but WITHOUT ANY WARRANTY; without even the implied warranty of\n",
    "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n",
    "GNU General Public License for more details.\n",
    "You should have received a copy of the GNU General Public License\n",
    "along with this program.  If not, see <http://www.gnu.org/licenses/>."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This Jupyter notebook analyzes 332 yeast genomes to find enzymes that act on carbohydrates\n",
    "\n",
    "Analysis performed by Martin Engqvist (Chalmers University of Technology) in November 2019.\n",
    "\n",
    "\n",
    "For the relative paths to work (which rely on the dotenv module) an empty file named exactly \".env\" must be placed in the project base folders. If this is not done all the folder variables below will need to be defined manually."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from dotenv import load_dotenv, find_dotenv # do 'pip install python-dotenv'\n",
    "from os.path import join, dirname, basename, exists, isdir\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import subprocess\n",
    "\n",
    "import Bio\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "### Load environmental variables from the project root directory ###\n",
    "# find .env automagically by walking up directories until it's found\n",
    "dotenv_path = find_dotenv()\n",
    "\n",
    "# load up the entries as environment variables\n",
    "load_dotenv(dotenv_path)\n",
    "\n",
    "# now you can get the variables using their names\n",
    "\n",
    "# Check whether a network drive has been specified\n",
    "DATABASE = os.environ.get(\"NETWORK_URL\")\n",
    "if DATABASE == 'None':\n",
    "    pass\n",
    "else:\n",
    "    pass\n",
    "    #mount network drive here\n",
    "\n",
    "# set up directory paths\n",
    "CURRENT_DIR = os.getcwd()\n",
    "PROJ = dirname(dotenv_path) # project root directory\n",
    "\n",
    "DATA = join(PROJ, 'data') #data directory\n",
    "RAW_EXTERNAL = join(DATA, 'raw_external') # external data raw directory\n",
    "RAW_INTERNAL = join(DATA, 'raw_internal') # internal data raw directory\n",
    "INTERMEDIATE = join(DATA, 'intermediate') # intermediate data directory\n",
    "FINAL = join(DATA, 'final') # final data directory\n",
    "\n",
    "RESULTS = join(PROJ, 'results') # output directory\n",
    "FIGURES = join(RESULTS, 'figures') # figure output directory\n",
    "PICTURES = join(RESULTS, 'pictures') # picture output directory\n",
    "   \n",
    "\n",
    "\n",
    "    \n",
    "tempo_clustered_path = join(INTERMEDIATE, 'tempo_in_genome', 'fasta', '332_genome_annotations', 'pep_clustered')\n",
    "if not exists(tempo_clustered_path):\n",
    "    os.makedirs(tempo_clustered_path)  \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python version: 3.7.7 (default, Sep 11 2020, 20:43:12) \n",
      "[GCC 7.3.0]\n",
      "Pandas version: 1.0.3\n",
      "Re version: 2.2.1\n",
      "BioPython version: 1.76\n"
     ]
    }
   ],
   "source": [
    "print('Python version: %s' % (sys.version))\n",
    "print('Pandas version: %s' % pd.__version__)\n",
    "print('Re version: %s' % re.__version__)\n",
    "print('BioPython version: %s' % Bio.__version__)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Notes\n",
    "The fasta files used in this script have been pre-processed in the \"de-duplication_of_genes.ipynb\" notebook. The pre-processing removed all whitespace from the fasta headers and subsequently clustered the sequences using cd-hit in order to remove duplicate sequences within a genome."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Start by loading genome information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>old_species_id</th>\n",
       "      <th>Species name</th>\n",
       "      <th>Common synonym</th>\n",
       "      <th>Major clade</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Strain sequenced</th>\n",
       "      <th>NRRL</th>\n",
       "      <th>CBS</th>\n",
       "      <th>...</th>\n",
       "      <th>GC%</th>\n",
       "      <th>Heterozygosity</th>\n",
       "      <th>CUG usage</th>\n",
       "      <th>BUSCO Complete</th>\n",
       "      <th>BUSCO Duplicated</th>\n",
       "      <th>BUSCO Fragmented</th>\n",
       "      <th>BUSCO Missing</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Note</th>\n",
       "      <th>organism</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Lipomyces_starkeyi</td>\n",
       "      <td>Lipomyces starkeyi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomyces</td>\n",
       "      <td>NRRL Y-11557</td>\n",
       "      <td>Y-11557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>46.92</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1566</td>\n",
       "      <td>11</td>\n",
       "      <td>126</td>\n",
       "      <td>56</td>\n",
       "      <td>Riley R, et al. Comparative genomics of biotec...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipomyces_starkeyi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>yHMPu5000034758_Lipomyces_japonicus</td>\n",
       "      <td>Lipomyces japonicus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomyces</td>\n",
       "      <td>NRRL Y-17848</td>\n",
       "      <td>Y-17848</td>\n",
       "      <td>7319.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.91</td>\n",
       "      <td>5.7E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1539</td>\n",
       "      <td>6</td>\n",
       "      <td>134</td>\n",
       "      <td>80</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipomyces_japonicus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>Tortispora_caseinolytica</td>\n",
       "      <td>Tortispora caseinolytica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Tortispora</td>\n",
       "      <td>NRRL Y-17796</td>\n",
       "      <td>Y-17796</td>\n",
       "      <td>7781.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.42</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1532</td>\n",
       "      <td>0</td>\n",
       "      <td>134</td>\n",
       "      <td>93</td>\n",
       "      <td>Riley R, et al. Comparative genomics of biotec...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tortispora_caseinolytica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>yHMPu5000035650_Trigonopsis_variabilis</td>\n",
       "      <td>Trigonopsis variabilis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsis</td>\n",
       "      <td>NRRL Y-1579</td>\n",
       "      <td>Y-1579</td>\n",
       "      <td>1040.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.28</td>\n",
       "      <td>1.3E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1308</td>\n",
       "      <td>3</td>\n",
       "      <td>262</td>\n",
       "      <td>186</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>trigonopsis_variabilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16</td>\n",
       "      <td>Arxula_adeninivorans</td>\n",
       "      <td>Blastobotrys adeninivorans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Trichomonascaceae</td>\n",
       "      <td>Blastobotrys</td>\n",
       "      <td>LS3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>48.12</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1663</td>\n",
       "      <td>7</td>\n",
       "      <td>75</td>\n",
       "      <td>14</td>\n",
       "      <td>Kunze G, et al. The complete genome of Blastob...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>blastobotrys_adeninivorans</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   id                          old_species_id                Species name  \\\n",
       "0   1                      Lipomyces_starkeyi          Lipomyces starkeyi   \n",
       "1   4     yHMPu5000034758_Lipomyces_japonicus         Lipomyces japonicus   \n",
       "2  10                Tortispora_caseinolytica    Tortispora caseinolytica   \n",
       "3  12  yHMPu5000035650_Trigonopsis_variabilis      Trigonopsis variabilis   \n",
       "4  16                    Arxula_adeninivorans  Blastobotrys adeninivorans   \n",
       "\n",
       "  Common synonym                      Major clade             Family  \\\n",
       "0            NaN                   Lipomycetaceae     Lipomycetaceae   \n",
       "1            NaN                   Lipomycetaceae     Lipomycetaceae   \n",
       "2            NaN                 Trigonopsidaceae   Trigonopsidaceae   \n",
       "3            NaN                 Trigonopsidaceae   Trigonopsidaceae   \n",
       "4            NaN  Dipodascaceae/Trichomonascaceae  Trichomonascaceae   \n",
       "\n",
       "          Genus Strain sequenced     NRRL     CBS  ...    GC% Heterozygosity  \\\n",
       "0     Lipomyces     NRRL Y-11557  Y-11557     NaN  ...  46.92        Unknown   \n",
       "1     Lipomyces     NRRL Y-17848  Y-17848  7319.0  ...  41.91        5.7E-05   \n",
       "2    Tortispora     NRRL Y-17796  Y-17796  7781.0  ...  45.42        Unknown   \n",
       "3   Trigonopsis      NRRL Y-1579   Y-1579  1040.0  ...  45.28        1.3E-05   \n",
       "4  Blastobotrys              LS3      NaN     NaN  ...  48.12        Unknown   \n",
       "\n",
       "  CUG usage BUSCO Complete BUSCO Duplicated BUSCO Fragmented  BUSCO Missing  \\\n",
       "0       Leu           1566               11              126             56   \n",
       "1       Leu           1539                6              134             80   \n",
       "2       Leu           1532                0              134             93   \n",
       "3       Leu           1308                3              262            186   \n",
       "4       Leu           1663                7               75             14   \n",
       "\n",
       "                                           Reference Note  \\\n",
       "0  Riley R, et al. Comparative genomics of biotec...  NaN   \n",
       "1                                         This study  NaN   \n",
       "2  Riley R, et al. Comparative genomics of biotec...  NaN   \n",
       "3                                         This study  NaN   \n",
       "4  Kunze G, et al. The complete genome of Blastob...  NaN   \n",
       "\n",
       "                     organism  \n",
       "0          lipomyces_starkeyi  \n",
       "1         lipomyces_japonicus  \n",
       "2    tortispora_caseinolytica  \n",
       "3      trigonopsis_variabilis  \n",
       "4  blastobotrys_adeninivorans  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>old_species_id</th>\n",
       "      <th>Species name</th>\n",
       "      <th>Common synonym</th>\n",
       "      <th>Major clade</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Strain sequenced</th>\n",
       "      <th>NRRL</th>\n",
       "      <th>CBS</th>\n",
       "      <th>...</th>\n",
       "      <th>GC%</th>\n",
       "      <th>Heterozygosity</th>\n",
       "      <th>CUG usage</th>\n",
       "      <th>BUSCO Complete</th>\n",
       "      <th>BUSCO Duplicated</th>\n",
       "      <th>BUSCO Fragmented</th>\n",
       "      <th>BUSCO Missing</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Note</th>\n",
       "      <th>organism</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>328</td>\n",
       "      <td>yHMPu5000034862_Zygotorulaspora_florentina</td>\n",
       "      <td>Zygotorulaspora florentina</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Zygotorulaspora</td>\n",
       "      <td>NRRL Y-1560</td>\n",
       "      <td>Y-1560</td>\n",
       "      <td>746.0</td>\n",
       "      <td>...</td>\n",
       "      <td>40.97</td>\n",
       "      <td>2.8E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1702</td>\n",
       "      <td>5</td>\n",
       "      <td>39</td>\n",
       "      <td>13</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zygotorulaspora_florentina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>329</td>\n",
       "      <td>yHMPu5000034710_Kluyveromyces_dobzhanskii</td>\n",
       "      <td>Kluyveromyces dobzhanskii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Kluyveromyces</td>\n",
       "      <td>NRRL Y-1974</td>\n",
       "      <td>Y-1974</td>\n",
       "      <td>2104.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.25</td>\n",
       "      <td>2.4E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1698</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>18</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kluyveromyces_dobzhanskii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>330</td>\n",
       "      <td>yHMPu5000034709_Kluyveromyces_aestuarii</td>\n",
       "      <td>Kluyveromyces aestuarii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Kluyveromyces</td>\n",
       "      <td>NRRL YB-4510</td>\n",
       "      <td>YB-4510</td>\n",
       "      <td>4438.0</td>\n",
       "      <td>...</td>\n",
       "      <td>38.32</td>\n",
       "      <td>2.2E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1671</td>\n",
       "      <td>2</td>\n",
       "      <td>55</td>\n",
       "      <td>31</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kluyveromyces_aestuarii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>331</td>\n",
       "      <td>yHMPu5000026256_Zygotorulaspora_mrakii</td>\n",
       "      <td>Zygotorulaspora mrakii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Zygotorulaspora</td>\n",
       "      <td>NRRL Y-12654</td>\n",
       "      <td>Y-12654</td>\n",
       "      <td>4218.0</td>\n",
       "      <td>...</td>\n",
       "      <td>39.90</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1432</td>\n",
       "      <td>3</td>\n",
       "      <td>35</td>\n",
       "      <td>289</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>332</td>\n",
       "      <td>yHMPu5000026152_Torulaspora_franciscae</td>\n",
       "      <td>Torulaspora franciscae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Torulaspora</td>\n",
       "      <td>NRRL Y-17532</td>\n",
       "      <td>Y-17532</td>\n",
       "      <td>2926.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.05</td>\n",
       "      <td>1E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1714</td>\n",
       "      <td>8</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>torulaspora_franciscae</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                              old_species_id  \\\n",
       "327  328  yHMPu5000034862_Zygotorulaspora_florentina   \n",
       "328  329   yHMPu5000034710_Kluyveromyces_dobzhanskii   \n",
       "329  330     yHMPu5000034709_Kluyveromyces_aestuarii   \n",
       "330  331      yHMPu5000026256_Zygotorulaspora_mrakii   \n",
       "331  332      yHMPu5000026152_Torulaspora_franciscae   \n",
       "\n",
       "                   Species name Common synonym         Major clade  \\\n",
       "327  Zygotorulaspora florentina            NaN  Saccharomycetaceae   \n",
       "328   Kluyveromyces dobzhanskii            NaN  Saccharomycetaceae   \n",
       "329     Kluyveromyces aestuarii            NaN  Saccharomycetaceae   \n",
       "330      Zygotorulaspora mrakii            NaN  Saccharomycetaceae   \n",
       "331      Torulaspora franciscae            NaN  Saccharomycetaceae   \n",
       "\n",
       "                 Family            Genus Strain sequenced     NRRL     CBS  \\\n",
       "327  Saccharomycetaceae  Zygotorulaspora      NRRL Y-1560   Y-1560   746.0   \n",
       "328  Saccharomycetaceae    Kluyveromyces      NRRL Y-1974   Y-1974  2104.0   \n",
       "329  Saccharomycetaceae    Kluyveromyces     NRRL YB-4510  YB-4510  4438.0   \n",
       "330  Saccharomycetaceae  Zygotorulaspora     NRRL Y-12654  Y-12654  4218.0   \n",
       "331  Saccharomycetaceae      Torulaspora     NRRL Y-17532  Y-17532  2926.0   \n",
       "\n",
       "     ...    GC% Heterozygosity CUG usage BUSCO Complete BUSCO Duplicated  \\\n",
       "327  ...  40.97        2.8E-05       Leu           1702                5   \n",
       "328  ...  41.25        2.4E-05       Leu           1698                7   \n",
       "329  ...  38.32        2.2E-05       Leu           1671                2   \n",
       "330  ...  39.90        Unknown       Leu           1432                3   \n",
       "331  ...  45.05          1E-05       Leu           1714                8   \n",
       "\n",
       "    BUSCO Fragmented  BUSCO Missing   Reference Note  \\\n",
       "327               39             13  This study  NaN   \n",
       "328               36             18  This study  NaN   \n",
       "329               55             31  This study  NaN   \n",
       "330               35            289  This study  NaN   \n",
       "331               29              8  This study  NaN   \n",
       "\n",
       "                       organism  \n",
       "327  zygotorulaspora_florentina  \n",
       "328   kluyveromyces_dobzhanskii  \n",
       "329     kluyveromyces_aestuarii  \n",
       "330      zygotorulaspora_mrakii  \n",
       "331      torulaspora_franciscae  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>CBS</th>\n",
       "      <th>JCM</th>\n",
       "      <th>No. contigs/scaffolds</th>\n",
       "      <th>No. genes</th>\n",
       "      <th>Assembly size</th>\n",
       "      <th>N50</th>\n",
       "      <th>GC%</th>\n",
       "      <th>BUSCO Complete</th>\n",
       "      <th>BUSCO Duplicated</th>\n",
       "      <th>BUSCO Fragmented</th>\n",
       "      <th>BUSCO Missing</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>332.000000</td>\n",
       "      <td>278.000000</td>\n",
       "      <td>138.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>3.320000e+02</td>\n",
       "      <td>3.320000e+02</td>\n",
       "      <td>332.0000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>166.500000</td>\n",
       "      <td>6424.077734</td>\n",
       "      <td>6761.065217</td>\n",
       "      <td>762.060241</td>\n",
       "      <td>5700.885542</td>\n",
       "      <td>1.293650e+07</td>\n",
       "      <td>7.523597e+05</td>\n",
       "      <td>40.5900</td>\n",
       "      <td>1606.445783</td>\n",
       "      <td>19.475904</td>\n",
       "      <td>70.801205</td>\n",
       "      <td>62.277108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>95.984374</td>\n",
       "      <td>3665.903382</td>\n",
       "      <td>6064.816152</td>\n",
       "      <td>2056.990772</td>\n",
       "      <td>1043.295863</td>\n",
       "      <td>3.195237e+06</td>\n",
       "      <td>7.257416e+05</td>\n",
       "      <td>5.6630</td>\n",
       "      <td>148.537991</td>\n",
       "      <td>34.418730</td>\n",
       "      <td>51.235930</td>\n",
       "      <td>114.334915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>107.000000</td>\n",
       "      <td>1442.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>4162.000000</td>\n",
       "      <td>8.152740e+06</td>\n",
       "      <td>2.738000e+03</td>\n",
       "      <td>25.2000</td>\n",
       "      <td>739.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>4.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>83.750000</td>\n",
       "      <td>2967.250000</td>\n",
       "      <td>1989.250000</td>\n",
       "      <td>39.250000</td>\n",
       "      <td>5060.000000</td>\n",
       "      <td>1.095436e+07</td>\n",
       "      <td>2.044210e+05</td>\n",
       "      <td>36.3450</td>\n",
       "      <td>1592.750000</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>42.000000</td>\n",
       "      <td>14.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>166.500000</td>\n",
       "      <td>6539.500000</td>\n",
       "      <td>3733.500000</td>\n",
       "      <td>307.500000</td>\n",
       "      <td>5556.000000</td>\n",
       "      <td>1.209838e+07</td>\n",
       "      <td>5.117265e+05</td>\n",
       "      <td>41.1400</td>\n",
       "      <td>1648.500000</td>\n",
       "      <td>8.500000</td>\n",
       "      <td>62.000000</td>\n",
       "      <td>24.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>249.250000</td>\n",
       "      <td>8925.500000</td>\n",
       "      <td>9771.250000</td>\n",
       "      <td>716.000000</td>\n",
       "      <td>5983.500000</td>\n",
       "      <td>1.405018e+07</td>\n",
       "      <td>1.121786e+06</td>\n",
       "      <td>44.8225</td>\n",
       "      <td>1686.250000</td>\n",
       "      <td>14.250000</td>\n",
       "      <td>85.000000</td>\n",
       "      <td>50.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>332.000000</td>\n",
       "      <td>14088.000000</td>\n",
       "      <td>31691.000000</td>\n",
       "      <td>28196.000000</td>\n",
       "      <td>12786.000000</td>\n",
       "      <td>2.583133e+07</td>\n",
       "      <td>4.956242e+06</td>\n",
       "      <td>53.9800</td>\n",
       "      <td>1735.000000</td>\n",
       "      <td>390.000000</td>\n",
       "      <td>526.000000</td>\n",
       "      <td>953.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               id           CBS           JCM  No. contigs/scaffolds  \\\n",
       "count  332.000000    278.000000    138.000000             332.000000   \n",
       "mean   166.500000   6424.077734   6761.065217             762.060241   \n",
       "std     95.984374   3665.903382   6064.816152            2056.990772   \n",
       "min      1.000000    107.000000   1442.000000               4.000000   \n",
       "25%     83.750000   2967.250000   1989.250000              39.250000   \n",
       "50%    166.500000   6539.500000   3733.500000             307.500000   \n",
       "75%    249.250000   8925.500000   9771.250000             716.000000   \n",
       "max    332.000000  14088.000000  31691.000000           28196.000000   \n",
       "\n",
       "          No. genes  Assembly size           N50       GC%  BUSCO Complete  \\\n",
       "count    332.000000   3.320000e+02  3.320000e+02  332.0000      332.000000   \n",
       "mean    5700.885542   1.293650e+07  7.523597e+05   40.5900     1606.445783   \n",
       "std     1043.295863   3.195237e+06  7.257416e+05    5.6630      148.537991   \n",
       "min     4162.000000   8.152740e+06  2.738000e+03   25.2000      739.000000   \n",
       "25%     5060.000000   1.095436e+07  2.044210e+05   36.3450     1592.750000   \n",
       "50%     5556.000000   1.209838e+07  5.117265e+05   41.1400     1648.500000   \n",
       "75%     5983.500000   1.405018e+07  1.121786e+06   44.8225     1686.250000   \n",
       "max    12786.000000   2.583133e+07  4.956242e+06   53.9800     1735.000000   \n",
       "\n",
       "       BUSCO Duplicated  BUSCO Fragmented  BUSCO Missing  \n",
       "count        332.000000        332.000000     332.000000  \n",
       "mean          19.475904         70.801205      62.277108  \n",
       "std           34.418730         51.235930     114.334915  \n",
       "min            0.000000         12.000000       4.000000  \n",
       "25%            5.000000         42.000000      14.000000  \n",
       "50%            8.500000         62.000000      24.000000  \n",
       "75%           14.250000         85.000000      50.000000  \n",
       "max          390.000000        526.000000     953.000000  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "def norm_organism(orgname):\n",
    "    '''\n",
    "    Return abbreviated organism name\n",
    "    '''\n",
    "    orgname = orgname.replace('[', '').replace(']', '')\n",
    "    \n",
    "    # two organisms occur twice, deal with these special cases\n",
    "    if orgname == 'Metschnikowia matae var. maris':\n",
    "        return 'metschnikowia_matae_maris'\n",
    "    \n",
    "    elif orgname == 'Nadsonia fulvescens var. elongata':\n",
    "        return 'nadsonia_fulvescens_var_elongata'\n",
    "      \n",
    "    # now parse filename for the others\n",
    "    if orgname.startswith('yH'):\n",
    "        organism = '_'.join(orgname.split()[1:3]).lower().replace('.', '')\n",
    "    else:\n",
    "        organism = '_'.join(orgname.split()[:2]).lower().replace('.', '')\n",
    "    \n",
    "    return organism\n",
    "\n",
    "\n",
    "\n",
    "def norm_filename(filename):\n",
    "    '''\n",
    "    Return abbreviated organism name\n",
    "    '''\n",
    "    filename = filename.replace('[', '').replace(']', '')\n",
    "    \n",
    "    # two organisms occur twice, deal with these special cases\n",
    "    if filename == 'metschnikowia_matae_maris.max.clustered.pep':\n",
    "        return 'metschnikowia_matae_maris'\n",
    "    \n",
    "    elif filename == 'nadsonia_fulvescens_var_elongata.max.clustered.pep':\n",
    "        return 'nadsonia_fulvescens_var_elongata'\n",
    "    \n",
    "    # now parse filename for the others\n",
    "    if filename.startswith('yH'):\n",
    "        file = '_'.join(filename.split('.')[0].split('_')[1:3]).lower()\n",
    "    else:\n",
    "        file = '_'.join(filename.split('.')[0].split('_')[:2]).lower()\n",
    "    \n",
    "    return file\n",
    "\n",
    "    \n",
    "\n",
    "# load up the data\n",
    "genome_data = pd.read_csv(join(PROJ, 'doc', '332_yeast_genomes.csv'), sep='\\t')\n",
    "\n",
    "# normalize the organism names\n",
    "genome_data['organism'] = genome_data.apply(lambda row: norm_organism(row['Species name']),axis=1)\n",
    "\n",
    "display(genome_data.head())\n",
    "display(genome_data.tail())\n",
    "display(genome_data.describe())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get genomes from tempo and mode of genome evolution paper\n",
    "\n",
    "Paper doi: 10.1016/j.cell.2018.10.023\n",
    "\n",
    "Data doi: 10.6084/m9.figshare.5854692\n",
    "\n",
    "Figshare link: https://figshare.com/articles/Tempo_and_mode_of_genome_evolution_in_the_budding_yeast_subphylum/5854692"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Orgs with tempo genomes:  306\n",
      "Orgs without genomes:  26\n"
     ]
    }
   ],
   "source": [
    "# collecte info for which file is where\n",
    "all_info = {}\n",
    "\n",
    "\n",
    "# get a list of all organisms\n",
    "all_orgs = set([norm_organism(s) for s in genome_data['organism'].tolist()])\n",
    "\n",
    "def get_from_tempo_files(wanted_orgs):\n",
    "    \n",
    "    filepath = tempo_clustered_path\n",
    "    files = os.listdir(filepath)\n",
    "    \n",
    "    orgs_found = set([])\n",
    "    for fi in files:\n",
    "        \n",
    "        if fi.endswith('.pep') or fi.endswith('.fasta'):\n",
    "            \n",
    "            norm_org = norm_filename(fi)\n",
    "\n",
    "            if norm_org in wanted_orgs:\n",
    "                orgs_found.add(norm_org)\n",
    "\n",
    "                all_info[norm_org] = {}\n",
    "                all_info[norm_org]['fasta'] = join(filepath, fi)\n",
    "            \n",
    "    return orgs_found\n",
    "\n",
    "\n",
    "\n",
    "tempo_orgs_found = get_from_tempo_files(all_orgs)\n",
    "\n",
    "tempo_orgs_not_found = all_orgs - tempo_orgs_found\n",
    "\n",
    "print('Orgs with tempo genomes: ', len(tempo_orgs_found))\n",
    "print('Orgs without genomes: ', len(tempo_orgs_not_found))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Not found\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>old_species_id</th>\n",
       "      <th>Species name</th>\n",
       "      <th>Common synonym</th>\n",
       "      <th>Major clade</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Strain sequenced</th>\n",
       "      <th>NRRL</th>\n",
       "      <th>CBS</th>\n",
       "      <th>...</th>\n",
       "      <th>GC%</th>\n",
       "      <th>Heterozygosity</th>\n",
       "      <th>CUG usage</th>\n",
       "      <th>BUSCO Complete</th>\n",
       "      <th>BUSCO Duplicated</th>\n",
       "      <th>BUSCO Fragmented</th>\n",
       "      <th>BUSCO Missing</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Note</th>\n",
       "      <th>organism</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Lipomyces_starkeyi</td>\n",
       "      <td>Lipomyces starkeyi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomyces</td>\n",
       "      <td>NRRL Y-11557</td>\n",
       "      <td>Y-11557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>46.92</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1566</td>\n",
       "      <td>11</td>\n",
       "      <td>126</td>\n",
       "      <td>56</td>\n",
       "      <td>Riley R, et al. Comparative genomics of biotec...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipomyces_starkeyi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>yHMPu5000034758_Lipomyces_japonicus</td>\n",
       "      <td>Lipomyces japonicus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomycetaceae</td>\n",
       "      <td>Lipomyces</td>\n",
       "      <td>NRRL Y-17848</td>\n",
       "      <td>Y-17848</td>\n",
       "      <td>7319.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.91</td>\n",
       "      <td>5.7E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1539</td>\n",
       "      <td>6</td>\n",
       "      <td>134</td>\n",
       "      <td>80</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipomyces_japonicus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>Tortispora_caseinolytica</td>\n",
       "      <td>Tortispora caseinolytica</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Tortispora</td>\n",
       "      <td>NRRL Y-17796</td>\n",
       "      <td>Y-17796</td>\n",
       "      <td>7781.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.42</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1532</td>\n",
       "      <td>0</td>\n",
       "      <td>134</td>\n",
       "      <td>93</td>\n",
       "      <td>Riley R, et al. Comparative genomics of biotec...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tortispora_caseinolytica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>yHMPu5000035650_Trigonopsis_variabilis</td>\n",
       "      <td>Trigonopsis variabilis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsidaceae</td>\n",
       "      <td>Trigonopsis</td>\n",
       "      <td>NRRL Y-1579</td>\n",
       "      <td>Y-1579</td>\n",
       "      <td>1040.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.28</td>\n",
       "      <td>1.3E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1308</td>\n",
       "      <td>3</td>\n",
       "      <td>262</td>\n",
       "      <td>186</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>trigonopsis_variabilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>24</td>\n",
       "      <td>Sugiyamaella_lignohabitans</td>\n",
       "      <td>Sugiyamaella lignohabitans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Trichomonascaceae</td>\n",
       "      <td>Sugiyamaella</td>\n",
       "      <td>CBS 10342</td>\n",
       "      <td>YB-1473</td>\n",
       "      <td>10342.0</td>\n",
       "      <td>...</td>\n",
       "      <td>44.86</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1654</td>\n",
       "      <td>9</td>\n",
       "      <td>83</td>\n",
       "      <td>13</td>\n",
       "      <td>Bellasio M, et al. Complete genome sequence an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sugiyamaella_lignohabitans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>328</td>\n",
       "      <td>yHMPu5000034862_Zygotorulaspora_florentina</td>\n",
       "      <td>Zygotorulaspora florentina</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Zygotorulaspora</td>\n",
       "      <td>NRRL Y-1560</td>\n",
       "      <td>Y-1560</td>\n",
       "      <td>746.0</td>\n",
       "      <td>...</td>\n",
       "      <td>40.97</td>\n",
       "      <td>2.8E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1702</td>\n",
       "      <td>5</td>\n",
       "      <td>39</td>\n",
       "      <td>13</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zygotorulaspora_florentina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>329</td>\n",
       "      <td>yHMPu5000034710_Kluyveromyces_dobzhanskii</td>\n",
       "      <td>Kluyveromyces dobzhanskii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Kluyveromyces</td>\n",
       "      <td>NRRL Y-1974</td>\n",
       "      <td>Y-1974</td>\n",
       "      <td>2104.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.25</td>\n",
       "      <td>2.4E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1698</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>18</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kluyveromyces_dobzhanskii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>330</td>\n",
       "      <td>yHMPu5000034709_Kluyveromyces_aestuarii</td>\n",
       "      <td>Kluyveromyces aestuarii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Kluyveromyces</td>\n",
       "      <td>NRRL YB-4510</td>\n",
       "      <td>YB-4510</td>\n",
       "      <td>4438.0</td>\n",
       "      <td>...</td>\n",
       "      <td>38.32</td>\n",
       "      <td>2.2E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1671</td>\n",
       "      <td>2</td>\n",
       "      <td>55</td>\n",
       "      <td>31</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kluyveromyces_aestuarii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>331</td>\n",
       "      <td>yHMPu5000026256_Zygotorulaspora_mrakii</td>\n",
       "      <td>Zygotorulaspora mrakii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Zygotorulaspora</td>\n",
       "      <td>NRRL Y-12654</td>\n",
       "      <td>Y-12654</td>\n",
       "      <td>4218.0</td>\n",
       "      <td>...</td>\n",
       "      <td>39.90</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1432</td>\n",
       "      <td>3</td>\n",
       "      <td>35</td>\n",
       "      <td>289</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>332</td>\n",
       "      <td>yHMPu5000026152_Torulaspora_franciscae</td>\n",
       "      <td>Torulaspora franciscae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Torulaspora</td>\n",
       "      <td>NRRL Y-17532</td>\n",
       "      <td>Y-17532</td>\n",
       "      <td>2926.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.05</td>\n",
       "      <td>1E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1714</td>\n",
       "      <td>8</td>\n",
       "      <td>29</td>\n",
       "      <td>8</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>torulaspora_franciscae</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>306 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                              old_species_id  \\\n",
       "0      1                          Lipomyces_starkeyi   \n",
       "1      4         yHMPu5000034758_Lipomyces_japonicus   \n",
       "2     10                    Tortispora_caseinolytica   \n",
       "3     12      yHMPu5000035650_Trigonopsis_variabilis   \n",
       "6     24                  Sugiyamaella_lignohabitans   \n",
       "..   ...                                         ...   \n",
       "327  328  yHMPu5000034862_Zygotorulaspora_florentina   \n",
       "328  329   yHMPu5000034710_Kluyveromyces_dobzhanskii   \n",
       "329  330     yHMPu5000034709_Kluyveromyces_aestuarii   \n",
       "330  331      yHMPu5000026256_Zygotorulaspora_mrakii   \n",
       "331  332      yHMPu5000026152_Torulaspora_franciscae   \n",
       "\n",
       "                   Species name Common synonym  \\\n",
       "0            Lipomyces starkeyi            NaN   \n",
       "1           Lipomyces japonicus            NaN   \n",
       "2      Tortispora caseinolytica            NaN   \n",
       "3        Trigonopsis variabilis            NaN   \n",
       "6    Sugiyamaella lignohabitans            NaN   \n",
       "..                          ...            ...   \n",
       "327  Zygotorulaspora florentina            NaN   \n",
       "328   Kluyveromyces dobzhanskii            NaN   \n",
       "329     Kluyveromyces aestuarii            NaN   \n",
       "330      Zygotorulaspora mrakii            NaN   \n",
       "331      Torulaspora franciscae            NaN   \n",
       "\n",
       "                         Major clade              Family            Genus  \\\n",
       "0                     Lipomycetaceae      Lipomycetaceae        Lipomyces   \n",
       "1                     Lipomycetaceae      Lipomycetaceae        Lipomyces   \n",
       "2                   Trigonopsidaceae    Trigonopsidaceae       Tortispora   \n",
       "3                   Trigonopsidaceae    Trigonopsidaceae      Trigonopsis   \n",
       "6    Dipodascaceae/Trichomonascaceae   Trichomonascaceae     Sugiyamaella   \n",
       "..                               ...                 ...              ...   \n",
       "327               Saccharomycetaceae  Saccharomycetaceae  Zygotorulaspora   \n",
       "328               Saccharomycetaceae  Saccharomycetaceae    Kluyveromyces   \n",
       "329               Saccharomycetaceae  Saccharomycetaceae    Kluyveromyces   \n",
       "330               Saccharomycetaceae  Saccharomycetaceae  Zygotorulaspora   \n",
       "331               Saccharomycetaceae  Saccharomycetaceae      Torulaspora   \n",
       "\n",
       "    Strain sequenced     NRRL      CBS  ...    GC% Heterozygosity CUG usage  \\\n",
       "0       NRRL Y-11557  Y-11557      NaN  ...  46.92        Unknown       Leu   \n",
       "1       NRRL Y-17848  Y-17848   7319.0  ...  41.91        5.7E-05       Leu   \n",
       "2       NRRL Y-17796  Y-17796   7781.0  ...  45.42        Unknown       Leu   \n",
       "3        NRRL Y-1579   Y-1579   1040.0  ...  45.28        1.3E-05       Leu   \n",
       "6          CBS 10342  YB-1473  10342.0  ...  44.86        Unknown       Leu   \n",
       "..               ...      ...      ...  ...    ...            ...       ...   \n",
       "327      NRRL Y-1560   Y-1560    746.0  ...  40.97        2.8E-05       Leu   \n",
       "328      NRRL Y-1974   Y-1974   2104.0  ...  41.25        2.4E-05       Leu   \n",
       "329     NRRL YB-4510  YB-4510   4438.0  ...  38.32        2.2E-05       Leu   \n",
       "330     NRRL Y-12654  Y-12654   4218.0  ...  39.90        Unknown       Leu   \n",
       "331     NRRL Y-17532  Y-17532   2926.0  ...  45.05          1E-05       Leu   \n",
       "\n",
       "    BUSCO Complete BUSCO Duplicated BUSCO Fragmented  BUSCO Missing  \\\n",
       "0             1566               11              126             56   \n",
       "1             1539                6              134             80   \n",
       "2             1532                0              134             93   \n",
       "3             1308                3              262            186   \n",
       "6             1654                9               83             13   \n",
       "..             ...              ...              ...            ...   \n",
       "327           1702                5               39             13   \n",
       "328           1698                7               36             18   \n",
       "329           1671                2               55             31   \n",
       "330           1432                3               35            289   \n",
       "331           1714                8               29              8   \n",
       "\n",
       "                                             Reference Note  \\\n",
       "0    Riley R, et al. Comparative genomics of biotec...  NaN   \n",
       "1                                           This study  NaN   \n",
       "2    Riley R, et al. Comparative genomics of biotec...  NaN   \n",
       "3                                           This study  NaN   \n",
       "6    Bellasio M, et al. Complete genome sequence an...  NaN   \n",
       "..                                                 ...  ...   \n",
       "327                                         This study  NaN   \n",
       "328                                         This study  NaN   \n",
       "329                                         This study  NaN   \n",
       "330                                         This study  NaN   \n",
       "331                                         This study  NaN   \n",
       "\n",
       "                       organism  \n",
       "0            lipomyces_starkeyi  \n",
       "1           lipomyces_japonicus  \n",
       "2      tortispora_caseinolytica  \n",
       "3        trigonopsis_variabilis  \n",
       "6    sugiyamaella_lignohabitans  \n",
       "..                          ...  \n",
       "327  zygotorulaspora_florentina  \n",
       "328   kluyveromyces_dobzhanskii  \n",
       "329     kluyveromyces_aestuarii  \n",
       "330      zygotorulaspora_mrakii  \n",
       "331      torulaspora_franciscae  \n",
       "\n",
       "[306 rows x 31 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>old_species_id</th>\n",
       "      <th>Species name</th>\n",
       "      <th>Common synonym</th>\n",
       "      <th>Major clade</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Strain sequenced</th>\n",
       "      <th>NRRL</th>\n",
       "      <th>CBS</th>\n",
       "      <th>...</th>\n",
       "      <th>GC%</th>\n",
       "      <th>Heterozygosity</th>\n",
       "      <th>CUG usage</th>\n",
       "      <th>BUSCO Complete</th>\n",
       "      <th>BUSCO Duplicated</th>\n",
       "      <th>BUSCO Fragmented</th>\n",
       "      <th>BUSCO Missing</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Note</th>\n",
       "      <th>organism</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16</td>\n",
       "      <td>Arxula_adeninivorans</td>\n",
       "      <td>Blastobotrys adeninivorans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Trichomonascaceae</td>\n",
       "      <td>Blastobotrys</td>\n",
       "      <td>LS3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>48.12</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1663</td>\n",
       "      <td>7</td>\n",
       "      <td>75</td>\n",
       "      <td>14</td>\n",
       "      <td>Kunze G, et al. The complete genome of Blastob...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>blastobotrys_adeninivorans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>17</td>\n",
       "      <td>Candida_apicola</td>\n",
       "      <td>Starmerella apicola</td>\n",
       "      <td>Candida apicola</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Dipodascaceae</td>\n",
       "      <td>Candida</td>\n",
       "      <td>NRRL Y-50540</td>\n",
       "      <td>Y-50540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>45.23</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1374</td>\n",
       "      <td>7</td>\n",
       "      <td>143</td>\n",
       "      <td>235</td>\n",
       "      <td>Vega-Alvarado L, et al. High-Quality Draft Gen...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>starmerella_apicola</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>68</td>\n",
       "      <td>Dekkera_bruxellensis</td>\n",
       "      <td>Brettanomyces bruxellensis</td>\n",
       "      <td>Dekkera bruxellensis</td>\n",
       "      <td>Pichiaceae</td>\n",
       "      <td>Pichiaceae</td>\n",
       "      <td>Brettanomyces</td>\n",
       "      <td>CBS 2499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2499.0</td>\n",
       "      <td>...</td>\n",
       "      <td>39.94</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1516</td>\n",
       "      <td>27</td>\n",
       "      <td>121</td>\n",
       "      <td>95</td>\n",
       "      <td>Piškur J, et al. The genome of wine yeast Dekk...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>brettanomyces_bruxellensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>101</td>\n",
       "      <td>yHMPu5000034637_Ogataea_populiabae</td>\n",
       "      <td>Ogataea populialbae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pichiaceae</td>\n",
       "      <td>Pichiaceae</td>\n",
       "      <td>Ogataea</td>\n",
       "      <td>NRRL Y-48632</td>\n",
       "      <td>Y-48632</td>\n",
       "      <td>11363.0</td>\n",
       "      <td>...</td>\n",
       "      <td>42.23</td>\n",
       "      <td>1.2E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1623</td>\n",
       "      <td>9</td>\n",
       "      <td>82</td>\n",
       "      <td>45</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ogataea_populialbae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>127</td>\n",
       "      <td>Candida_homilentoma</td>\n",
       "      <td>Hyphopichia homilentoma</td>\n",
       "      <td>Candida homilentoma</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Hyphopichia</td>\n",
       "      <td>JCM 1507</td>\n",
       "      <td>Y-10941</td>\n",
       "      <td>6312.0</td>\n",
       "      <td>...</td>\n",
       "      <td>49.52</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1702</td>\n",
       "      <td>4</td>\n",
       "      <td>48</td>\n",
       "      <td>5</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hyphopichia_homilentoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>133</td>\n",
       "      <td>Candida_tenuis</td>\n",
       "      <td>Yamadazyma tenuis</td>\n",
       "      <td>Candida tenuis</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Yamadazyma</td>\n",
       "      <td>NRRL Y-1498</td>\n",
       "      <td>Y-1498</td>\n",
       "      <td>615.0</td>\n",
       "      <td>...</td>\n",
       "      <td>42.92</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1681</td>\n",
       "      <td>8</td>\n",
       "      <td>58</td>\n",
       "      <td>12</td>\n",
       "      <td>Wohlbach DJ, et al. Comparative genomics of xy...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>yamadazyma_tenuis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>200</td>\n",
       "      <td>yHMPu5000035031_Candida_kruisii</td>\n",
       "      <td>Teunomyces kruisii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Teunomyces</td>\n",
       "      <td>NRRL Y-17087</td>\n",
       "      <td>Y-17087</td>\n",
       "      <td>6451.0</td>\n",
       "      <td>...</td>\n",
       "      <td>46.07</td>\n",
       "      <td>1E-06</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1689</td>\n",
       "      <td>36</td>\n",
       "      <td>29</td>\n",
       "      <td>5</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>teunomyces_kruisii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>201</td>\n",
       "      <td>yHMPu5000035022_Candida_emberorum</td>\n",
       "      <td>Suhomyces emberorum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Suhomyces</td>\n",
       "      <td>NRRL Y-27606</td>\n",
       "      <td>Y-27606</td>\n",
       "      <td>9827.0</td>\n",
       "      <td>...</td>\n",
       "      <td>40.75</td>\n",
       "      <td>3E-06</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1696</td>\n",
       "      <td>5</td>\n",
       "      <td>46</td>\n",
       "      <td>12</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>suhomyces_emberorum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>18</td>\n",
       "      <td>Candida_infanticola</td>\n",
       "      <td>Wickerhamiella infanticola</td>\n",
       "      <td>Candida infanticola</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Dipodascaceae</td>\n",
       "      <td>Wickerhamiella</td>\n",
       "      <td>DS02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>48.32</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1477</td>\n",
       "      <td>10</td>\n",
       "      <td>147</td>\n",
       "      <td>125</td>\n",
       "      <td>Lee H, et al. Genbank Accession IDs: LWLF01000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wickerhamiella_infanticola</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>19</td>\n",
       "      <td>Candida_versatilis</td>\n",
       "      <td>Wickerhamiella versatilis</td>\n",
       "      <td>Candida versatilis</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Dipodascaceae</td>\n",
       "      <td>Wickerhamiella</td>\n",
       "      <td>JCM 5958</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7511.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.04</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1437</td>\n",
       "      <td>7</td>\n",
       "      <td>158</td>\n",
       "      <td>157</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>wickerhamiella_versatilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>30</td>\n",
       "      <td>yHMPu5000041818_Magnusiomyces_tetrasperma</td>\n",
       "      <td>Magnusiomyces tetraspermus</td>\n",
       "      <td>Magnusiomyces tetrasperma</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Dipodascaceae</td>\n",
       "      <td>Magnusiomyces</td>\n",
       "      <td>NRRL Y-7288</td>\n",
       "      <td>Y-7288</td>\n",
       "      <td>765.7</td>\n",
       "      <td>...</td>\n",
       "      <td>38.23</td>\n",
       "      <td>8.2E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>909</td>\n",
       "      <td>229</td>\n",
       "      <td>159</td>\n",
       "      <td>462</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>magnusiomyces_tetraspermus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>31</td>\n",
       "      <td>yHMPu5000035691_Candida_salmanticensis</td>\n",
       "      <td>Groenewaldozyma salmanticensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Saccharomycetales incertae sedis</td>\n",
       "      <td>Groenewaldozyma</td>\n",
       "      <td>NRRL Y-17090</td>\n",
       "      <td>Y-17090</td>\n",
       "      <td>5121.0</td>\n",
       "      <td>...</td>\n",
       "      <td>43.35</td>\n",
       "      <td>4E-06</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1625</td>\n",
       "      <td>8</td>\n",
       "      <td>99</td>\n",
       "      <td>27</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>groenewaldozyma_salmanticensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>46</td>\n",
       "      <td>yHMPu5000034669_Blastobotrys_raffinofermentans</td>\n",
       "      <td>Blastobotrys raffinosifermentans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dipodascaceae/Trichomonascaceae</td>\n",
       "      <td>Trichomonascaceae</td>\n",
       "      <td>Blastobotrys</td>\n",
       "      <td>NRRL Y-27150</td>\n",
       "      <td>Y-27150</td>\n",
       "      <td>6800.0</td>\n",
       "      <td>...</td>\n",
       "      <td>48.22</td>\n",
       "      <td>0</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1659</td>\n",
       "      <td>16</td>\n",
       "      <td>70</td>\n",
       "      <td>14</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>blastobotrys_raffinosifermentans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>94</td>\n",
       "      <td>yHMPu5000034893_Ogataea_philodendra</td>\n",
       "      <td>Ogataea philodendri</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pichiaceae</td>\n",
       "      <td>Pichiaceae</td>\n",
       "      <td>Ogataea</td>\n",
       "      <td>NRRL Y-7210</td>\n",
       "      <td>Y-7210</td>\n",
       "      <td>6075.0</td>\n",
       "      <td>...</td>\n",
       "      <td>47.85</td>\n",
       "      <td>8E-06</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1625</td>\n",
       "      <td>8</td>\n",
       "      <td>85</td>\n",
       "      <td>41</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ogataea_philodendri</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>132</td>\n",
       "      <td>Candida_tanzawaensis</td>\n",
       "      <td>Suhomyces tanzawaensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Suhomyces</td>\n",
       "      <td>NRRL Y-17324</td>\n",
       "      <td>Y-17324</td>\n",
       "      <td>7422.0</td>\n",
       "      <td>...</td>\n",
       "      <td>45.34</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1719</td>\n",
       "      <td>5</td>\n",
       "      <td>31</td>\n",
       "      <td>4</td>\n",
       "      <td>Riley R, et al. Comparative genomics of biotec...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>suhomyces_tanzawaensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>146</td>\n",
       "      <td>Metschnikowia_dekortum</td>\n",
       "      <td>Metschnikowia dekortorum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Metschnikowiaceae</td>\n",
       "      <td>Metschnikowia</td>\n",
       "      <td>UWOPS 01-142b3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>48.73</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1660</td>\n",
       "      <td>7</td>\n",
       "      <td>72</td>\n",
       "      <td>20</td>\n",
       "      <td>Lachance MA, et al. A stable phylogeny of the ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metschnikowia_dekortorum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>154</td>\n",
       "      <td>Metschnikowia_lockheadii</td>\n",
       "      <td>Metschnikowia lochheadii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Metschnikowiaceae</td>\n",
       "      <td>Metschnikowia</td>\n",
       "      <td>UWOPS 99-661.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>42.16</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1663</td>\n",
       "      <td>5</td>\n",
       "      <td>70</td>\n",
       "      <td>21</td>\n",
       "      <td>Lachance MA, et al. A stable phylogeny of the ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metschnikowia_lochheadii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>198</td>\n",
       "      <td>yHMPu5000035033_Candida_cretensis</td>\n",
       "      <td>Teunomyces cretensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Teunomyces</td>\n",
       "      <td>NRRL Y-27777</td>\n",
       "      <td>Y-27777</td>\n",
       "      <td>9453.0</td>\n",
       "      <td>...</td>\n",
       "      <td>46.16</td>\n",
       "      <td>1E-06</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1721</td>\n",
       "      <td>7</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>teunomyces_cretensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>199</td>\n",
       "      <td>yHMPu5000035032_Candida_gatunensis</td>\n",
       "      <td>Teunomyces gatunensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Teunomyces</td>\n",
       "      <td>NRRL Y-48064</td>\n",
       "      <td>Y-48064</td>\n",
       "      <td>10379.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.86</td>\n",
       "      <td>1.1E-05</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1704</td>\n",
       "      <td>6</td>\n",
       "      <td>36</td>\n",
       "      <td>13</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>teunomyces_gatunensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>202</td>\n",
       "      <td>yHMPu5000035018_Candida_canberraensis</td>\n",
       "      <td>Suhomyces canberraensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Suhomyces</td>\n",
       "      <td>NRRL YB-2417</td>\n",
       "      <td>YB-2417</td>\n",
       "      <td>8846.0</td>\n",
       "      <td>...</td>\n",
       "      <td>40.01</td>\n",
       "      <td>2E-06</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1710</td>\n",
       "      <td>8</td>\n",
       "      <td>32</td>\n",
       "      <td>9</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>suhomyces_canberraensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>203</td>\n",
       "      <td>yHMPu5000035011_Candida_pyralidae</td>\n",
       "      <td>Suhomyces pyralidae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CUG-Ser1</td>\n",
       "      <td>Debaryomycetaceae</td>\n",
       "      <td>Suhomyces</td>\n",
       "      <td>NRRL Y-27085</td>\n",
       "      <td>Y-27085</td>\n",
       "      <td>5035.0</td>\n",
       "      <td>...</td>\n",
       "      <td>42.67</td>\n",
       "      <td>8E-06</td>\n",
       "      <td>Ser</td>\n",
       "      <td>1710</td>\n",
       "      <td>2</td>\n",
       "      <td>40</td>\n",
       "      <td>7</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "      <td>suhomyces_pyralidae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>238</td>\n",
       "      <td>yHMPu5000035286_Candida_azyma</td>\n",
       "      <td>Wickerhamomyces sp.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Phaffomycetaceae</td>\n",
       "      <td>Phaffomycetaceae</td>\n",
       "      <td>Wickerhamomyces</td>\n",
       "      <td>NRRL YB-2243</td>\n",
       "      <td>YB-2243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>37.42</td>\n",
       "      <td>1.5E-05</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1669</td>\n",
       "      <td>9</td>\n",
       "      <td>52</td>\n",
       "      <td>29</td>\n",
       "      <td>This study</td>\n",
       "      <td>The \"old_species_id\" column and some supplemen...</td>\n",
       "      <td>wickerhamomyces_sp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>256</td>\n",
       "      <td>Hanseniaspora_vinae</td>\n",
       "      <td>Hanseniaspora vineae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycodaceae</td>\n",
       "      <td>Saccharomycodaceae</td>\n",
       "      <td>Hanseniaspora</td>\n",
       "      <td>T02/19AF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>37.41</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1508</td>\n",
       "      <td>1</td>\n",
       "      <td>112</td>\n",
       "      <td>138</td>\n",
       "      <td>Giorello FM, et al. Genome Sequence of the Nat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hanseniaspora_vineae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>285</td>\n",
       "      <td>Nakaseomyces_bracarensis</td>\n",
       "      <td>Candida bracarensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Candida</td>\n",
       "      <td>CBS 10154</td>\n",
       "      <td>Y-48270</td>\n",
       "      <td>10154.0</td>\n",
       "      <td>...</td>\n",
       "      <td>36.13</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1647</td>\n",
       "      <td>51</td>\n",
       "      <td>39</td>\n",
       "      <td>22</td>\n",
       "      <td>Gabaldon T, et al. Comparative genomics of eme...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>candida_bracarensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>286</td>\n",
       "      <td>Nakaseomyces_castellii</td>\n",
       "      <td>Candida castellii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Candida</td>\n",
       "      <td>CBS 4332</td>\n",
       "      <td>Y-17070</td>\n",
       "      <td>4332.0</td>\n",
       "      <td>...</td>\n",
       "      <td>40.86</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1601</td>\n",
       "      <td>30</td>\n",
       "      <td>78</td>\n",
       "      <td>50</td>\n",
       "      <td>Gabaldon T, et al. Comparative genomics of eme...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>candida_castellii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>288</td>\n",
       "      <td>Nakaseomyces_nivariensis</td>\n",
       "      <td>Candida nivariensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Saccharomycetaceae</td>\n",
       "      <td>Candida</td>\n",
       "      <td>CBS 9983</td>\n",
       "      <td>Y-48269</td>\n",
       "      <td>9983.0</td>\n",
       "      <td>...</td>\n",
       "      <td>36.86</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Leu</td>\n",
       "      <td>1648</td>\n",
       "      <td>55</td>\n",
       "      <td>37</td>\n",
       "      <td>19</td>\n",
       "      <td>Gabaldon T, et al. Comparative genomics of eme...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>candida_nivariensis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      id                                  old_species_id  \\\n",
       "4     16                            Arxula_adeninivorans   \n",
       "5     17                                 Candida_apicola   \n",
       "20    68                            Dekkera_bruxellensis   \n",
       "32   101              yHMPu5000034637_Ogataea_populiabae   \n",
       "40   127                             Candida_homilentoma   \n",
       "43   133                                  Candida_tenuis   \n",
       "63   200                 yHMPu5000035031_Candida_kruisii   \n",
       "64   201               yHMPu5000035022_Candida_emberorum   \n",
       "111   18                             Candida_infanticola   \n",
       "112   19                              Candida_versatilis   \n",
       "121   30       yHMPu5000041818_Magnusiomyces_tetrasperma   \n",
       "122   31          yHMPu5000035691_Candida_salmanticensis   \n",
       "132   46  yHMPu5000034669_Blastobotrys_raffinofermentans   \n",
       "161   94             yHMPu5000034893_Ogataea_philodendra   \n",
       "188  132                            Candida_tanzawaensis   \n",
       "196  146                          Metschnikowia_dekortum   \n",
       "203  154                        Metschnikowia_lockheadii   \n",
       "234  198               yHMPu5000035033_Candida_cretensis   \n",
       "235  199              yHMPu5000035032_Candida_gatunensis   \n",
       "236  202           yHMPu5000035018_Candida_canberraensis   \n",
       "237  203               yHMPu5000035011_Candida_pyralidae   \n",
       "262  238                   yHMPu5000035286_Candida_azyma   \n",
       "276  256                             Hanseniaspora_vinae   \n",
       "296  285                        Nakaseomyces_bracarensis   \n",
       "297  286                          Nakaseomyces_castellii   \n",
       "299  288                        Nakaseomyces_nivariensis   \n",
       "\n",
       "                         Species name             Common synonym  \\\n",
       "4          Blastobotrys adeninivorans                        NaN   \n",
       "5                 Starmerella apicola            Candida apicola   \n",
       "20         Brettanomyces bruxellensis       Dekkera bruxellensis   \n",
       "32                Ogataea populialbae                        NaN   \n",
       "40            Hyphopichia homilentoma        Candida homilentoma   \n",
       "43                  Yamadazyma tenuis             Candida tenuis   \n",
       "63                 Teunomyces kruisii                        NaN   \n",
       "64                Suhomyces emberorum                        NaN   \n",
       "111        Wickerhamiella infanticola        Candida infanticola   \n",
       "112         Wickerhamiella versatilis         Candida versatilis   \n",
       "121        Magnusiomyces tetraspermus  Magnusiomyces tetrasperma   \n",
       "122    Groenewaldozyma salmanticensis                        NaN   \n",
       "132  Blastobotrys raffinosifermentans                        NaN   \n",
       "161               Ogataea philodendri                        NaN   \n",
       "188            Suhomyces tanzawaensis                        NaN   \n",
       "196          Metschnikowia dekortorum                        NaN   \n",
       "203          Metschnikowia lochheadii                        NaN   \n",
       "234              Teunomyces cretensis                        NaN   \n",
       "235             Teunomyces gatunensis                        NaN   \n",
       "236           Suhomyces canberraensis                        NaN   \n",
       "237               Suhomyces pyralidae                        NaN   \n",
       "262               Wickerhamomyces sp.                        NaN   \n",
       "276              Hanseniaspora vineae                        NaN   \n",
       "296               Candida bracarensis                        NaN   \n",
       "297                 Candida castellii                        NaN   \n",
       "299               Candida nivariensis                        NaN   \n",
       "\n",
       "                         Major clade                            Family  \\\n",
       "4    Dipodascaceae/Trichomonascaceae                 Trichomonascaceae   \n",
       "5    Dipodascaceae/Trichomonascaceae                     Dipodascaceae   \n",
       "20                        Pichiaceae                        Pichiaceae   \n",
       "32                        Pichiaceae                        Pichiaceae   \n",
       "40                          CUG-Ser1                 Debaryomycetaceae   \n",
       "43                          CUG-Ser1                 Debaryomycetaceae   \n",
       "63                          CUG-Ser1                 Debaryomycetaceae   \n",
       "64                          CUG-Ser1                 Debaryomycetaceae   \n",
       "111  Dipodascaceae/Trichomonascaceae                     Dipodascaceae   \n",
       "112  Dipodascaceae/Trichomonascaceae                     Dipodascaceae   \n",
       "121  Dipodascaceae/Trichomonascaceae                     Dipodascaceae   \n",
       "122  Dipodascaceae/Trichomonascaceae  Saccharomycetales incertae sedis   \n",
       "132  Dipodascaceae/Trichomonascaceae                 Trichomonascaceae   \n",
       "161                       Pichiaceae                        Pichiaceae   \n",
       "188                         CUG-Ser1                 Debaryomycetaceae   \n",
       "196                         CUG-Ser1                 Metschnikowiaceae   \n",
       "203                         CUG-Ser1                 Metschnikowiaceae   \n",
       "234                         CUG-Ser1                 Debaryomycetaceae   \n",
       "235                         CUG-Ser1                 Debaryomycetaceae   \n",
       "236                         CUG-Ser1                 Debaryomycetaceae   \n",
       "237                         CUG-Ser1                 Debaryomycetaceae   \n",
       "262                 Phaffomycetaceae                  Phaffomycetaceae   \n",
       "276               Saccharomycodaceae                Saccharomycodaceae   \n",
       "296               Saccharomycetaceae                Saccharomycetaceae   \n",
       "297               Saccharomycetaceae                Saccharomycetaceae   \n",
       "299               Saccharomycetaceae                Saccharomycetaceae   \n",
       "\n",
       "               Genus Strain sequenced     NRRL      CBS  ...    GC%  \\\n",
       "4       Blastobotrys              LS3      NaN      NaN  ...  48.12   \n",
       "5            Candida     NRRL Y-50540  Y-50540      NaN  ...  45.23   \n",
       "20     Brettanomyces         CBS 2499      NaN   2499.0  ...  39.94   \n",
       "32           Ogataea     NRRL Y-48632  Y-48632  11363.0  ...  42.23   \n",
       "40       Hyphopichia         JCM 1507  Y-10941   6312.0  ...  49.52   \n",
       "43        Yamadazyma      NRRL Y-1498   Y-1498    615.0  ...  42.92   \n",
       "63        Teunomyces     NRRL Y-17087  Y-17087   6451.0  ...  46.07   \n",
       "64         Suhomyces     NRRL Y-27606  Y-27606   9827.0  ...  40.75   \n",
       "111   Wickerhamiella             DS02      NaN      NaN  ...  48.32   \n",
       "112   Wickerhamiella         JCM 5958      NaN   7511.0  ...  45.04   \n",
       "121    Magnusiomyces      NRRL Y-7288   Y-7288    765.7  ...  38.23   \n",
       "122  Groenewaldozyma     NRRL Y-17090  Y-17090   5121.0  ...  43.35   \n",
       "132     Blastobotrys     NRRL Y-27150  Y-27150   6800.0  ...  48.22   \n",
       "161          Ogataea      NRRL Y-7210   Y-7210   6075.0  ...  47.85   \n",
       "188        Suhomyces     NRRL Y-17324  Y-17324   7422.0  ...  45.34   \n",
       "196    Metschnikowia   UWOPS 01-142b3      NaN      NaN  ...  48.73   \n",
       "203    Metschnikowia   UWOPS 99-661.1      NaN      NaN  ...  42.16   \n",
       "234       Teunomyces     NRRL Y-27777  Y-27777   9453.0  ...  46.16   \n",
       "235       Teunomyces     NRRL Y-48064  Y-48064  10379.0  ...  41.86   \n",
       "236        Suhomyces     NRRL YB-2417  YB-2417   8846.0  ...  40.01   \n",
       "237        Suhomyces     NRRL Y-27085  Y-27085   5035.0  ...  42.67   \n",
       "262  Wickerhamomyces     NRRL YB-2243  YB-2243      NaN  ...  37.42   \n",
       "276    Hanseniaspora         T02/19AF      NaN      NaN  ...  37.41   \n",
       "296          Candida        CBS 10154  Y-48270  10154.0  ...  36.13   \n",
       "297          Candida         CBS 4332  Y-17070   4332.0  ...  40.86   \n",
       "299          Candida         CBS 9983  Y-48269   9983.0  ...  36.86   \n",
       "\n",
       "    Heterozygosity CUG usage BUSCO Complete BUSCO Duplicated BUSCO Fragmented  \\\n",
       "4          Unknown       Leu           1663                7               75   \n",
       "5          Unknown       Leu           1374                7              143   \n",
       "20         Unknown       Leu           1516               27              121   \n",
       "32         1.2E-05       Leu           1623                9               82   \n",
       "40         Unknown       Ser           1702                4               48   \n",
       "43         Unknown       Ser           1681                8               58   \n",
       "63           1E-06       Ser           1689               36               29   \n",
       "64           3E-06       Ser           1696                5               46   \n",
       "111        Unknown       Leu           1477               10              147   \n",
       "112        Unknown       Leu           1437                7              158   \n",
       "121        8.2E-05       Leu            909              229              159   \n",
       "122          4E-06       Leu           1625                8               99   \n",
       "132              0       Leu           1659               16               70   \n",
       "161          8E-06       Leu           1625                8               85   \n",
       "188        Unknown       Ser           1719                5               31   \n",
       "196        Unknown       Ser           1660                7               72   \n",
       "203        Unknown       Ser           1663                5               70   \n",
       "234          1E-06       Ser           1721                7               26   \n",
       "235        1.1E-05       Ser           1704                6               36   \n",
       "236          2E-06       Ser           1710                8               32   \n",
       "237          8E-06       Ser           1710                2               40   \n",
       "262        1.5E-05       Leu           1669                9               52   \n",
       "276        Unknown       Leu           1508                1              112   \n",
       "296        Unknown       Leu           1647               51               39   \n",
       "297        Unknown       Leu           1601               30               78   \n",
       "299        Unknown       Leu           1648               55               37   \n",
       "\n",
       "     BUSCO Missing                                          Reference  \\\n",
       "4               14  Kunze G, et al. The complete genome of Blastob...   \n",
       "5              235  Vega-Alvarado L, et al. High-Quality Draft Gen...   \n",
       "20              95  Piškur J, et al. The genome of wine yeast Dekk...   \n",
       "32              45                                         This study   \n",
       "40               5                                         This study   \n",
       "43              12  Wohlbach DJ, et al. Comparative genomics of xy...   \n",
       "63               5                                         This study   \n",
       "64              12                                         This study   \n",
       "111            125  Lee H, et al. Genbank Accession IDs: LWLF01000...   \n",
       "112            157                                         This study   \n",
       "121            462                                         This study   \n",
       "122             27                                         This study   \n",
       "132             14                                         This study   \n",
       "161             41                                         This study   \n",
       "188              4  Riley R, et al. Comparative genomics of biotec...   \n",
       "196             20  Lachance MA, et al. A stable phylogeny of the ...   \n",
       "203             21  Lachance MA, et al. A stable phylogeny of the ...   \n",
       "234              5                                         This study   \n",
       "235             13                                         This study   \n",
       "236              9                                         This study   \n",
       "237              7                                         This study   \n",
       "262             29                                         This study   \n",
       "276            138  Giorello FM, et al. Genome Sequence of the Nat...   \n",
       "296             22  Gabaldon T, et al. Comparative genomics of eme...   \n",
       "297             50  Gabaldon T, et al. Comparative genomics of eme...   \n",
       "299             19  Gabaldon T, et al. Comparative genomics of eme...   \n",
       "\n",
       "                                                  Note  \\\n",
       "4                                                  NaN   \n",
       "5                                                  NaN   \n",
       "20                                                 NaN   \n",
       "32                                                 NaN   \n",
       "40                                                 NaN   \n",
       "43                                                 NaN   \n",
       "63                                                 NaN   \n",
       "64                                                 NaN   \n",
       "111                                                NaN   \n",
       "112                                                NaN   \n",
       "121                                                NaN   \n",
       "122                                                NaN   \n",
       "132                                                NaN   \n",
       "161                                                NaN   \n",
       "188                                                NaN   \n",
       "196                                                NaN   \n",
       "203                                                NaN   \n",
       "234                                                NaN   \n",
       "235                                                NaN   \n",
       "236                                                NaN   \n",
       "237                                                NaN   \n",
       "262  The \"old_species_id\" column and some supplemen...   \n",
       "276                                                NaN   \n",
       "296                                                NaN   \n",
       "297                                                NaN   \n",
       "299                                                NaN   \n",
       "\n",
       "                             organism  \n",
       "4          blastobotrys_adeninivorans  \n",
       "5                 starmerella_apicola  \n",
       "20         brettanomyces_bruxellensis  \n",
       "32                ogataea_populialbae  \n",
       "40            hyphopichia_homilentoma  \n",
       "43                  yamadazyma_tenuis  \n",
       "63                 teunomyces_kruisii  \n",
       "64                suhomyces_emberorum  \n",
       "111        wickerhamiella_infanticola  \n",
       "112         wickerhamiella_versatilis  \n",
       "121        magnusiomyces_tetraspermus  \n",
       "122    groenewaldozyma_salmanticensis  \n",
       "132  blastobotrys_raffinosifermentans  \n",
       "161               ogataea_philodendri  \n",
       "188            suhomyces_tanzawaensis  \n",
       "196          metschnikowia_dekortorum  \n",
       "203          metschnikowia_lochheadii  \n",
       "234              teunomyces_cretensis  \n",
       "235             teunomyces_gatunensis  \n",
       "236           suhomyces_canberraensis  \n",
       "237               suhomyces_pyralidae  \n",
       "262                wickerhamomyces_sp  \n",
       "276              hanseniaspora_vineae  \n",
       "296               candida_bracarensis  \n",
       "297                 candida_castellii  \n",
       "299               candida_nivariensis  \n",
       "\n",
       "[26 rows x 31 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "print('Not found')\n",
    "display(genome_data[genome_data['organism'].isin(tempo_orgs_found)])\n",
    "\n",
    "print('Found')\n",
    "display(genome_data[genome_data['organism'].isin(tempo_orgs_not_found)])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Seems like some are not found because the old species name is used on the fasta files in some cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Orgs with tempo genomes:  26\n",
      "Orgs without genomes:  0\n"
     ]
    }
   ],
   "source": [
    "\n",
    "missing_subset = genome_data[genome_data['organism'].isin(tempo_orgs_not_found)]\n",
    "\n",
    "all_orgs_alt_names = set([])\n",
    "for org in missing_subset['old_species_id'].tolist():\n",
    "    all_orgs_alt_names.add(norm_filename(org))\n",
    "        \n",
    "\n",
    "\n",
    "tempo2_orgs_found = get_from_tempo_files(all_orgs_alt_names)\n",
    "\n",
    "tempo2_orgs_not_found = all_orgs_alt_names - tempo2_orgs_found\n",
    "\n",
    "print('Orgs with tempo genomes: ', len(tempo2_orgs_found))\n",
    "print('Orgs without genomes: ', len(tempo2_orgs_not_found))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run dbCan HMMs on the fasta files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### First I need to download the HMMs\n",
    "\n",
    "http://bcb.unl.edu/dbCAN2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  /data/Work/projects/yeast_carbohydrate_enz/data/raw_external/dbcan/dbCAN-HMMdb-V8.zip\n",
      "  inflating: dbCAN-HMMdb-V8.txt      \n"
     ]
    }
   ],
   "source": [
    "\n",
    "hmm_folder = join(RAW_EXTERNAL, 'dbcan')\n",
    "hmm_file = join(hmm_folder, 'dbCAN-HMMdb-V8.txt')\n",
    "\n",
    "if not exists(hmm_folder):\n",
    "    os.makedirs(hmm_folder)\n",
    "\n",
    "if not exists(hmm_file):\n",
    "    # unzip the zipfile (which comes with the repository)\n",
    "    hmm_zip_file = hmm_file.replace('.txt', '.zip')\n",
    "    if exists(hmm_zip_file.replace('.txt', '.zip')):\n",
    "        !unzip $hmm_zip_file\n",
    "    \n",
    "    # if it is not present download the file again\n",
    "    else:\n",
    "        url = 'http://bcb.unl.edu/dbCAN2/download/dbCAN-HMMdb-V8.txt'\n",
    "        mycmd = 'wget %s -P %s' % (url, hmm_folder)\n",
    "        os.system(mycmd)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Run the HMM on all genome fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'# hmmsearch :: search profile(s) against a sequence database\\n# HMMER 3.3.1 (Jul 2020); http://hmmer.org/\\n# Copyright (C) 2020 Howard Hughes Medical Institute.\\n# Freely distributed under the BSD open source license.\\n'\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "# get version\n",
    "mycmd = 'hmmsearch -h | head -n 4'\n",
    "print(subprocess.check_output(mycmd, shell=True))\n",
    "\n",
    "# make folder if needed\n",
    "outpath = join(INTERMEDIATE, 'hmm_output')\n",
    "if not exists(outpath):\n",
    "    os.makedirs(outpath)\n",
    "\n",
    "\n",
    "# run hmmer on all fasta files    \n",
    "dbcan_hmms = join(RAW_EXTERNAL, 'dbcan', 'dbCAN-HMMdb-V8.txt')\n",
    "for org in sorted(all_info.keys()):\n",
    "    fasta_file = all_info[org]['fasta']\n",
    "    hmm_outfile = join(outpath, org+'.out')\n",
    "    \n",
    "    if not exists(hmm_outfile):\n",
    "    \n",
    "        mycmd = 'hmmsearch -E 1e-15 --domtblout %s %s %s > tmp.out' % (hmm_outfile, dbcan_hmms, fasta_file)\n",
    "        os.system(mycmd)\n",
    "\n",
    "    all_info[org]['hmm_out'] = hmm_outfile\n",
    "\n",
    "print('Done')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Run singalP on all genomes to get a signal peptide prediction\n",
    "\n",
    "http://www.cbs.dtu.dk/services/SignalP/\n",
    "\n",
    "SignalP version 5.0b Linux x86_64\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# make folder if needed\n",
    "outpath = join(INTERMEDIATE, 'signalp_output')\n",
    "if not exists(outpath):\n",
    "    os.makedirs(outpath)\n",
    "\n",
    "\n",
    "# run signalP on all genomes\n",
    "for org in sorted(all_info.keys()):\n",
    "    fasta_file = all_info[org]['fasta']\n",
    "    signalp_outfile = join(outpath, org+'.out')\n",
    "    \n",
    "    if not exists(signalp_outfile):\n",
    "    \n",
    "        mycmd = 'signalp -fasta %s -org euk -format short -stdout > %s' % (fasta_file, signalp_outfile)\n",
    "        os.system(mycmd)\n",
    "        \n",
    "    all_info[org]['signalp_out'] = signalp_outfile\n",
    "\n",
    "\n",
    "print('Done')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Parse the outfiles and combine to a single data file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism</th>\n",
       "      <th>gene</th>\n",
       "      <th>hmm_model</th>\n",
       "      <th>hmm_model_len</th>\n",
       "      <th>hmm_match_from</th>\n",
       "      <th>hmm_match_to</th>\n",
       "      <th>hmm_match_coverage</th>\n",
       "      <th>match_evalue</th>\n",
       "      <th>gene_match_from</th>\n",
       "      <th>gene_match_to</th>\n",
       "      <th>enzyme</th>\n",
       "      <th>family</th>\n",
       "      <th>enzyme_type</th>\n",
       "      <th>signal_peptide</th>\n",
       "      <th>signal_peptide_prob</th>\n",
       "      <th>sp_cut_pos</th>\n",
       "      <th>sp_cut_seq</th>\n",
       "      <th>sp_cut_prob</th>\n",
       "      <th>genes_in_fasta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>aciculoconidium_aculeatum</td>\n",
       "      <td>augustus_masked-NODE_13_length_332279_cov_34.6...</td>\n",
       "      <td>CBM21.hmm</td>\n",
       "      <td>107</td>\n",
       "      <td>2</td>\n",
       "      <td>107</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.8e-31</td>\n",
       "      <td>355</td>\n",
       "      <td>480</td>\n",
       "      <td>CBM21</td>\n",
       "      <td>CBM21</td>\n",
       "      <td>CBM</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>0.001684</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>6944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>aciculoconidium_aculeatum</td>\n",
       "      <td>snap_masked-NODE_98_length_34288_cov_31.3655_I...</td>\n",
       "      <td>CBM21.hmm</td>\n",
       "      <td>107</td>\n",
       "      <td>3</td>\n",
       "      <td>107</td>\n",
       "      <td>0.97</td>\n",
       "      <td>2.8e-22</td>\n",
       "      <td>265</td>\n",
       "      <td>371</td>\n",
       "      <td>CBM21</td>\n",
       "      <td>CBM21</td>\n",
       "      <td>CBM</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>0.000843</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>6944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>aciculoconidium_aculeatum</td>\n",
       "      <td>augustus_masked-NODE_42_length_144261_cov_33.5...</td>\n",
       "      <td>CBM43.hmm</td>\n",
       "      <td>83</td>\n",
       "      <td>1</td>\n",
       "      <td>82</td>\n",
       "      <td>0.98</td>\n",
       "      <td>6.4e-21</td>\n",
       "      <td>372</td>\n",
       "      <td>457</td>\n",
       "      <td>CBM43</td>\n",
       "      <td>CBM43</td>\n",
       "      <td>CBM</td>\n",
       "      <td>SP(Sec/SPI)</td>\n",
       "      <td>0.995878</td>\n",
       "      <td>19-20</td>\n",
       "      <td>VSA-AD</td>\n",
       "      <td>0.8042</td>\n",
       "      <td>6944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>aciculoconidium_aculeatum</td>\n",
       "      <td>augustus_masked-NODE_97_length_34420_cov_46.27...</td>\n",
       "      <td>CBM43.hmm</td>\n",
       "      <td>83</td>\n",
       "      <td>1</td>\n",
       "      <td>82</td>\n",
       "      <td>0.98</td>\n",
       "      <td>3e-20</td>\n",
       "      <td>380</td>\n",
       "      <td>465</td>\n",
       "      <td>CBM43</td>\n",
       "      <td>CBM43</td>\n",
       "      <td>CBM</td>\n",
       "      <td>SP(Sec/SPI)</td>\n",
       "      <td>0.922542</td>\n",
       "      <td>22-23</td>\n",
       "      <td>ALA-NN</td>\n",
       "      <td>0.6326</td>\n",
       "      <td>6944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>aciculoconidium_aculeatum</td>\n",
       "      <td>augustus_masked-NODE_3_length_592973_cov_34.37...</td>\n",
       "      <td>CE10.hmm</td>\n",
       "      <td>341</td>\n",
       "      <td>93</td>\n",
       "      <td>286</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.4e-19</td>\n",
       "      <td>96</td>\n",
       "      <td>322</td>\n",
       "      <td>CE10</td>\n",
       "      <td>CE10</td>\n",
       "      <td>CE</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>0.025548</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>6944</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    organism  \\\n",
       "0  aciculoconidium_aculeatum   \n",
       "1  aciculoconidium_aculeatum   \n",
       "2  aciculoconidium_aculeatum   \n",
       "3  aciculoconidium_aculeatum   \n",
       "4  aciculoconidium_aculeatum   \n",
       "\n",
       "                                                gene  hmm_model hmm_model_len  \\\n",
       "0  augustus_masked-NODE_13_length_332279_cov_34.6...  CBM21.hmm           107   \n",
       "1  snap_masked-NODE_98_length_34288_cov_31.3655_I...  CBM21.hmm           107   \n",
       "2  augustus_masked-NODE_42_length_144261_cov_33.5...  CBM43.hmm            83   \n",
       "3  augustus_masked-NODE_97_length_34420_cov_46.27...  CBM43.hmm            83   \n",
       "4  augustus_masked-NODE_3_length_592973_cov_34.37...   CE10.hmm           341   \n",
       "\n",
       "  hmm_match_from hmm_match_to hmm_match_coverage match_evalue gene_match_from  \\\n",
       "0              2          107               0.98      1.8e-31             355   \n",
       "1              3          107               0.97      2.8e-22             265   \n",
       "2              1           82               0.98      6.4e-21             372   \n",
       "3              1           82               0.98        3e-20             380   \n",
       "4             93          286               0.57      1.4e-19              96   \n",
       "\n",
       "  gene_match_to enzyme family enzyme_type signal_peptide signal_peptide_prob  \\\n",
       "0           480  CBM21  CBM21         CBM          OTHER            0.001684   \n",
       "1           371  CBM21  CBM21         CBM          OTHER            0.000843   \n",
       "2           457  CBM43  CBM43         CBM    SP(Sec/SPI)            0.995878   \n",
       "3           465  CBM43  CBM43         CBM    SP(Sec/SPI)            0.922542   \n",
       "4           322   CE10   CE10          CE          OTHER            0.025548   \n",
       "\n",
       "  sp_cut_pos sp_cut_seq sp_cut_prob  genes_in_fasta  \n",
       "0        N/A        N/A         N/A            6944  \n",
       "1        N/A        N/A         N/A            6944  \n",
       "2      19-20     VSA-AD      0.8042            6944  \n",
       "3      22-23     ALA-NN      0.6326            6944  \n",
       "4        N/A        N/A         N/A            6944  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism</th>\n",
       "      <th>gene</th>\n",
       "      <th>hmm_model</th>\n",
       "      <th>hmm_model_len</th>\n",
       "      <th>hmm_match_from</th>\n",
       "      <th>hmm_match_to</th>\n",
       "      <th>hmm_match_coverage</th>\n",
       "      <th>match_evalue</th>\n",
       "      <th>gene_match_from</th>\n",
       "      <th>gene_match_to</th>\n",
       "      <th>enzyme</th>\n",
       "      <th>family</th>\n",
       "      <th>enzyme_type</th>\n",
       "      <th>signal_peptide</th>\n",
       "      <th>signal_peptide_prob</th>\n",
       "      <th>sp_cut_pos</th>\n",
       "      <th>sp_cut_seq</th>\n",
       "      <th>sp_cut_prob</th>\n",
       "      <th>genes_in_fasta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "      <td>genemark-NODE_4_length_587279_cov_18.5498_ID_5...</td>\n",
       "      <td>GH5_54.hmm</td>\n",
       "      <td>263</td>\n",
       "      <td>2</td>\n",
       "      <td>236</td>\n",
       "      <td>0.89</td>\n",
       "      <td>5.4e-22</td>\n",
       "      <td>92</td>\n",
       "      <td>370</td>\n",
       "      <td>GH5_54</td>\n",
       "      <td>GH5</td>\n",
       "      <td>GH</td>\n",
       "      <td>SP(Sec/SPI)</td>\n",
       "      <td>0.934510</td>\n",
       "      <td>23-24</td>\n",
       "      <td>VNG-LD</td>\n",
       "      <td>0.9126</td>\n",
       "      <td>4757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "      <td>genemark-NODE_1_length_1100707_cov_17.8797_ID_...</td>\n",
       "      <td>GH5_54.hmm</td>\n",
       "      <td>263</td>\n",
       "      <td>9</td>\n",
       "      <td>255</td>\n",
       "      <td>0.94</td>\n",
       "      <td>3.6e-19</td>\n",
       "      <td>87</td>\n",
       "      <td>349</td>\n",
       "      <td>GH5_54</td>\n",
       "      <td>GH5</td>\n",
       "      <td>GH</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>0.001628</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>4757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "      <td>genemark-NODE_3_length_705464_cov_18.5159_ID_3...</td>\n",
       "      <td>GH5_9.hmm</td>\n",
       "      <td>303</td>\n",
       "      <td>2</td>\n",
       "      <td>303</td>\n",
       "      <td>0.99</td>\n",
       "      <td>3.4e-124</td>\n",
       "      <td>112</td>\n",
       "      <td>417</td>\n",
       "      <td>GH5_9</td>\n",
       "      <td>GH5</td>\n",
       "      <td>GH</td>\n",
       "      <td>SP(Sec/SPI)</td>\n",
       "      <td>0.900421</td>\n",
       "      <td>20-21</td>\n",
       "      <td>AVS-SP</td>\n",
       "      <td>0.5682</td>\n",
       "      <td>4757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "      <td>genemark-NODE_4_length_587279_cov_18.5498_ID_5...</td>\n",
       "      <td>GH5_9.hmm</td>\n",
       "      <td>303</td>\n",
       "      <td>1</td>\n",
       "      <td>303</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1e-113</td>\n",
       "      <td>107</td>\n",
       "      <td>451</td>\n",
       "      <td>GH5_9</td>\n",
       "      <td>GH5</td>\n",
       "      <td>GH</td>\n",
       "      <td>SP(Sec/SPI)</td>\n",
       "      <td>0.934510</td>\n",
       "      <td>23-24</td>\n",
       "      <td>VNG-LD</td>\n",
       "      <td>0.9126</td>\n",
       "      <td>4757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "      <td>genemark-NODE_1_length_1100707_cov_17.8797_ID_...</td>\n",
       "      <td>GH5_9.hmm</td>\n",
       "      <td>303</td>\n",
       "      <td>1</td>\n",
       "      <td>231</td>\n",
       "      <td>0.76</td>\n",
       "      <td>5.1e-35</td>\n",
       "      <td>94</td>\n",
       "      <td>331</td>\n",
       "      <td>GH5_9</td>\n",
       "      <td>GH5</td>\n",
       "      <td>GH</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>0.001628</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>N/A</td>\n",
       "      <td>4757</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   organism  \\\n",
       "189  zygotorulaspora_mrakii   \n",
       "190  zygotorulaspora_mrakii   \n",
       "191  zygotorulaspora_mrakii   \n",
       "192  zygotorulaspora_mrakii   \n",
       "193  zygotorulaspora_mrakii   \n",
       "\n",
       "                                                  gene   hmm_model  \\\n",
       "189  genemark-NODE_4_length_587279_cov_18.5498_ID_5...  GH5_54.hmm   \n",
       "190  genemark-NODE_1_length_1100707_cov_17.8797_ID_...  GH5_54.hmm   \n",
       "191  genemark-NODE_3_length_705464_cov_18.5159_ID_3...   GH5_9.hmm   \n",
       "192  genemark-NODE_4_length_587279_cov_18.5498_ID_5...   GH5_9.hmm   \n",
       "193  genemark-NODE_1_length_1100707_cov_17.8797_ID_...   GH5_9.hmm   \n",
       "\n",
       "    hmm_model_len hmm_match_from hmm_match_to hmm_match_coverage match_evalue  \\\n",
       "189           263              2          236               0.89      5.4e-22   \n",
       "190           263              9          255               0.94      3.6e-19   \n",
       "191           303              2          303               0.99     3.4e-124   \n",
       "192           303              1          303               1.00       1e-113   \n",
       "193           303              1          231               0.76      5.1e-35   \n",
       "\n",
       "    gene_match_from gene_match_to  enzyme family enzyme_type signal_peptide  \\\n",
       "189              92           370  GH5_54    GH5          GH    SP(Sec/SPI)   \n",
       "190              87           349  GH5_54    GH5          GH          OTHER   \n",
       "191             112           417   GH5_9    GH5          GH    SP(Sec/SPI)   \n",
       "192             107           451   GH5_9    GH5          GH    SP(Sec/SPI)   \n",
       "193              94           331   GH5_9    GH5          GH          OTHER   \n",
       "\n",
       "    signal_peptide_prob sp_cut_pos sp_cut_seq sp_cut_prob  genes_in_fasta  \n",
       "189            0.934510      23-24     VNG-LD      0.9126            4757  \n",
       "190            0.001628        N/A        N/A         N/A            4757  \n",
       "191            0.900421      20-21     AVS-SP      0.5682            4757  \n",
       "192            0.934510      23-24     VNG-LD      0.9126            4757  \n",
       "193            0.001628        N/A        N/A         N/A            4757  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genes_in_fasta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>85211.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>5774.603666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1061.953000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>4128.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>5111.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>5565.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>6131.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>11757.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       genes_in_fasta\n",
       "count    85211.000000\n",
       "mean      5774.603666\n",
       "std       1061.953000\n",
       "min       4128.000000\n",
       "25%       5111.000000\n",
       "50%       5565.000000\n",
       "75%       6131.000000\n",
       "max      11757.000000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "def parse_hmm_output(filepath, org):\n",
    "    '''\n",
    "    Parse an hmm output file and return as pandas data frame.\n",
    "    '''\n",
    "    data = {'organism':[], 'gene':[], 'hmm_model':[], 'hmm_model_len':[], 'hmm_match_from':[], \n",
    "            'hmm_match_to':[], 'hmm_match_coverage':[], 'match_evalue':[], 'gene_match_from':[], \n",
    "            'gene_match_to':[], 'enzyme':[], 'family':[], 'enzyme_type':[]}\n",
    "    \n",
    "    with open(filepath, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):\n",
    "                continue\n",
    "            \n",
    "            parts = line.strip().split()\n",
    "            gene = parts[0]\n",
    "            \n",
    "            hmm = parts[3]\n",
    "            full_enzyme = hmm.split('.')[0]\n",
    "            m = re.search('([A-Z]+)([0-9]+)[_]?([0-9]+)?', full_enzyme)\n",
    "            family = m[1]+m[2]\n",
    "            enzyme_type = m[1]\n",
    "\n",
    "            evalue = parts[6]\n",
    "            \n",
    "            qlen = parts[5]\n",
    "            hmm_from = parts[15]\n",
    "            hmm_to = parts[16]\n",
    "            coverage = (int(hmm_to)-int(hmm_from))/int(qlen)\n",
    "            \n",
    "            ali_from = parts[17]\n",
    "            ali_to = parts[18]\n",
    "            \n",
    "            if coverage >= 0.35:\n",
    "                data['organism'].append(org)\n",
    "                data['gene'].append(gene)\n",
    "                \n",
    "                data['hmm_model'].append(hmm)\n",
    "                data['hmm_model_len'].append(qlen)\n",
    "                data['hmm_match_from'].append(hmm_from)\n",
    "                data['hmm_match_to'].append(hmm_to)\n",
    "                data['hmm_match_coverage'].append('%.2f' % float(coverage))\n",
    "                data['match_evalue'].append(evalue)\n",
    "                data['gene_match_from'].append(ali_from)\n",
    "                data['gene_match_to'].append(ali_to)\n",
    "\n",
    "                data['enzyme'].append(full_enzyme)   \n",
    "                data['family'].append(family)            \n",
    "                data['enzyme_type'].append(enzyme_type)          \n",
    "   \n",
    "            \n",
    "    return pd.DataFrame(data)\n",
    "           \n",
    "\n",
    "    \n",
    "def parse_signalp_output(filepath, org):\n",
    "    '''\n",
    "    Parse an signalP output file and return as pandas data frame.\n",
    "    '''\n",
    "    data = {'organism':[], 'gene':[], 'signal_peptide':[], 'signal_peptide_prob':[],\n",
    "            'sp_cut_pos':[], 'sp_cut_seq':[], 'sp_cut_prob':[]}\n",
    "\n",
    "    with open(filepath, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):\n",
    "                continue\n",
    "\n",
    "            parts = line.strip().split('\\t')\n",
    "            gene = parts[0]\n",
    "            sp_or_not = parts[1]\n",
    "            pred_sp = parts[2]\n",
    "            pred_other = parts[3]\n",
    "\n",
    "            \n",
    "            if sp_or_not != 'OTHER':\n",
    "                \n",
    "                # find signal peptide cut-site info\n",
    "                m = re.search('CS pos: ([0-9]+[-][0-9]+)[.] ([A-Z]+[-][A-Z]+)[.] Pr: ([01]+[.][0-9]+)', parts[4])\n",
    "                \n",
    "                if m is None:\n",
    "                    cut_pos = 'N/A'\n",
    "                    cut_seq = 'N/A'\n",
    "                    cut_pval = 'N/A'\n",
    "                \n",
    "                else:\n",
    "                    cut_pos = m[1]\n",
    "                    cut_seq = m[2]\n",
    "                    cut_pval = m[3]\n",
    "                \n",
    "            else:\n",
    "                cut_pos = 'N/A'\n",
    "                cut_seq = 'N/A'\n",
    "                cut_pval = 'N/A'\n",
    "         \n",
    "            data['organism'].append(org)\n",
    "            data['gene'].append(gene)\n",
    "            data['signal_peptide'].append(sp_or_not)\n",
    "            data['signal_peptide_prob'].append(pred_sp)\n",
    "            data['sp_cut_pos'].append(cut_pos)\n",
    "            data['sp_cut_seq'].append(cut_seq)\n",
    "            data['sp_cut_prob'].append(cut_pval)\n",
    "\n",
    "    return pd.DataFrame(data)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "def get_fasta_gene_number(filepath, org):\n",
    "    '''\n",
    "    For an organism count the number of genes in fasta file.\n",
    "    '''\n",
    "    data = {'organism':[], 'genes_in_fasta':[]}\n",
    "    counter = 0\n",
    "    for record in SeqIO.parse(filepath, \"fasta\"):\n",
    "        counter += 1\n",
    "        \n",
    "    data['organism'].append(org)\n",
    "    data['genes_in_fasta'].append(counter)\n",
    "    \n",
    "    return pd.DataFrame(data)\n",
    "    \n",
    "\n",
    "        \n",
    "all_genome_data = None\n",
    "for organism in sorted(all_info.keys()):\n",
    "\n",
    "    # parse outputs\n",
    "    hmm_data = parse_hmm_output(all_info[organism]['hmm_out'], organism)\n",
    "    signalp_data = parse_signalp_output(all_info[organism]['signalp_out'], organism)\n",
    "    fasta_data = get_fasta_gene_number(all_info[organism]['fasta'], organism)\n",
    "\n",
    "    \n",
    "    # merge the data frames \n",
    "    hmm_data = hmm_data.merge(signalp_data, on=['organism', 'gene'], how='left')\n",
    "    hmm_data = hmm_data.merge(fasta_data, on='organism', how='left')\n",
    "    \n",
    "\n",
    "    # add to the main data \n",
    "    if all_genome_data is None:\n",
    "        all_genome_data = hmm_data\n",
    "    else:\n",
    "        all_genome_data = all_genome_data.append(hmm_data)\n",
    "\n",
    "        \n",
    "# apparently CAZy has removed GH10, so let's remove that from the data before saving\n",
    "all_genome_data = all_genome_data[all_genome_data.family!='GH10']\n",
    "    \n",
    "display(all_genome_data.head())\n",
    "display(all_genome_data.tail())\n",
    "display(all_genome_data.describe())\n",
    "\n",
    "# write file to disk\n",
    "all_genome_data.to_csv(join(FINAL, '332_yeast_genomes_enzyme_info_version_2.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### There are sometimes several predictions for the same region of a protein, but we only want to keep one."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### First figure out what the cutoff should be in terms of what is an overlap."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unfiltered domain number: 85211\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/martin/anaconda3/envs/py37/lib/python3.7/site-packages/ipykernel_launcher.py:105: MatplotlibDeprecationWarning: \n",
      "The frameon kwarg was deprecated in Matplotlib 3.1 and will be removed in 3.3. Use facecolor instead.\n",
      "/home/martin/anaconda3/envs/py37/lib/python3.7/site-packages/ipykernel_launcher.py:110: MatplotlibDeprecationWarning: \n",
      "The frameon kwarg was deprecated in Matplotlib 3.1 and will be removed in 3.3. Use facecolor instead.\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZEAAAEWCAYAAACnlKo3AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3de5gcVZ3/8feHhEuAQIAEJAkxoIEVkOsAQRSBVQKoEFdWYVGCoijLqog3WFHB24q4oKzKwg+Qi4QoSCAqELJIQJHbhFsSIBIhQhIkQQjXGEL4/v44p5PKpGemp2a6ZzrzeT1PP1N9qk6dU5epb9epqlOKCMzMzMpYp7crYGZmzctBxMzMSnMQMTOz0hxEzMysNAcRMzMrzUHEzMxKcxCxhpI0StLLkgb0dl06IukMSb/o7Xp0l6RjJN3cwfh3SZpT47xulDSh52pXH5JC0lu7kb8p9tG+Qn5OZO0naR6wFfA6sAJ4GLgcuDAi3ujFqvVZks4A3hoRH+3tuvQkSQGMiYi53ZzPccAnI+KdPVKxHtRTy2i18ZlI//GBiBgMvBn4PvBV4OLerVLfJGlgb9fBrFk4iPQzEfFCREwBPgJMkLQzgKRNJV0uabGkv0o6XdI6edxxku6QdK6kJZIel/SOnP6UpEXFZg5J75N0v6QX8/gzCuNG5+aGgfn7dEnfzvN/SdLNkoa2V39Jn5I0V9JzkqZIGp7T/1fSD9tMe72kU/LwcEm/zsv3hKTPFaY7Q9I1kn4h6UXguCrlXi3pb5JekHS7pJ0K4y7N5U/Ly3CbpDe3U/8Ncjl/z+vyXklbFbbBxZKelrRA0ncqTSp5Xf9R0g8lPZ+X4dDCfI/L2+WlPO6YYr48fHue/MHcXPMRSQdImp/Hnyrpmjb1/bGk8wrb6pOS3gb8L7Bvns8SSXtJeqYYgCV9SNID7ayHWvaRCZKelPSspK8Vxu8t6c5c7tOSfiJpvSpldFinPJ/WXIdnJJ3TpvzKPlp13VoWEf6s5R9gHvCeKulPAifm4cuB64HBwGjgz8DxedxxpKawjwMDgO/kvD8F1gcOBl4CNs7THwC8nfQjZRfgGWB8HjcaCGBg/j4d+AuwPTAof/9+O8txEPAssEcu93+A2/O4/YGnWNVEuxmwFBie6zED+AawHrAd8DgwLk97BrAcGJ+nHZTTflEo+xN53awP/Ah4oDDu0rz8++fxPwb+2M4yfBr4DbBhXpd7ApvkcdcBFwAbAVsC9wCfLmyD5cCncr4TgYWA8vQvAjvkabcGdirk+2Oh/CA101HYVvPz8JuBVwv1GQA8DYwtbKtPVptvTnsYOLTwfTLwxXbWwwF0vo/8v7wtdgWWAW/L4/cExgID87SPACdXW8aO6gTcCXwsD29cWM5K+QM7Wrf+5HXa2xXwpwEbuf0gchfwtXywWAbsWBj3aWB6Hj4OeKww7u35n2yrQtrfgd3aKf9HwLl5eOU/aP4+HTi9MO2/Aze1M5+LgR8Uvm9MOrCOJh1MnwT2z+M+Bfw+D+8DPNlmXqcBP8/DZ5CDUWH8GRSCSJtxQ/IybJq/XwpMalOvFcA2VfJ+AvgTsEub9K3yNhhUSDsauLWwDeYWxm2Y6/CmfKBbAnyomL+Qr6Ygkr//ETg2D78X+Eth3HQ6DiJfBa7Mw5uTAtLWNe6j1faRkYXx9wBHtZP3ZGBytWXsqE7A7cCZwNA286uUXwkiVdetP+nj5qz+bQTwHDCU9Av9r4Vxf83jK54pDC8FiIi2aRsDSNpH0q256egF4DO5jPb8rTD8amU+VQwv1jEiXiYFrxGR/vsnkQ68AP8GXJmH3wwMz80fSyQtAf6TdOCueKq9ykkaIOn7kv6Sm7vm5VHFZVqZP9fruVzftq4ApgKTJC2U9ANJ6+Y6rgs8XajjBaQzkoqV6ykiXs2DG0fEK6Tmyc/k/L+T9E/tLU8nJrL6OpzYhby/AD4gaWPgw8AfIuLpahPWuI9U3S8kbS/pt7l58UXge1Xy1lKn40lnwI/mZsX3t83cw+t2reQg0k9J2osUJP5IaiJaTjqQVYwCFpSc/URgCumX+Kak9nOVr+1KCynUUdJGwBasqudVwJH5esQ+wK9z+lPAExExpPAZHBGHFebd0W2K/wYcAbwH2JT0SxVWX6ZtCvXamPSrd2HbGUXE8og4MyJ2BN4BvB84NtdxGelXcaWOm0TETm3nUU1ETI2I95KaWx4lNQWVcTVwgKSRwAdpP4issb4iYgGpieiDwMdIAbM93dlHzict45iI2IT0g6Bq3o7qFBGPRcTRpEB9FnBN3qfazqOn1u1ayUGkn5G0Sf7FNYnUXDMzIlYAvwK+K2lwPgifQvoVV8Zg4LmI+IekvUkH4Z4wEfi4pN0krU/6BXp3RMwDiIj7gcXARcDUiFiS890DvCjpq5IG5TOLnXMgrXV5lpHOejbM5bZ1mKR35gu83871WuPsRtKBkt6eL5i/SAreK/Kv45uB/87baB1Jb5H07s4qJ2krSYfnA+Ay4GVSc1o1z5CuCVUVEYtJzVY/JwXeRzqYz8gqF7QvB75CavKc3EG1u7OPDCatu5fzWcGJnUxftU6SPippWKTb3Cv7ymrrrYvrtl9yEOk/fiPpJdIv3q8B55AulFd8FniFdMH5j6QD9iUly/p34Fu5vG+QAlS3RcQtwNdJZxhPA28Bjmoz2VWkM4aJhXwrgA8AuwFPkM68LiKdVdTiclIz2gLShdq7qkwzEfgmqRlrT6C9O3jeBFxDOgg+AtzGqmB9LKlZ8WHg+Tzd1jXUbx3gi6Qzn+eAd5O2QTVnAJflJrMPtzPNRNqswyp+D8wG/ibp2UL6ZNLZ4uTcFNSe7uwjXyIFnZdIZwW/7GT69up0CDBb0sukmyGOioh/tMnblXXbL/lhQ7NuknQp6eL06b1dl75A0l9Id5X9X2/XpaIv1mlt4TMRM+sxkj5Eul7y+96uS0VfrNPaxE/mmlmPkDQd2JH07EWf6E6nL9ZpbePmLDMzK83NWWZmVlq/a84aOnRojB49urerYWbWVGbMmPFsRAxrm97vgsjo0aNpbW3t7WqYmTUVSX+tlu7mLDMzK81BxMzMSnMQMTOz0hxEzMysNAcRMzMrrd/dnWVm1p9cd/8Czp46h4VLljJ8yCC+PG4Hxu8+ovOMNXIQMTNbS113/wJOu3YmS5en3usXLFnKadfOBOixQOLmLDOztdTZU+esDCAVS5ev4Oypc3qsDAcRM7O11MIlS7uUXoaDiJnZWmr4kEFdSi/DQcTMbC315XE7MGjdAaulDVp3AF8et0OPleEL62Zma6nKxXPfnWVmZqWM331EjwaNttycZWZmpTmImJlZaQ4iZmZWmoOImZmV5iBiZmalOYiYmVlpDiJmZlaag4iZmZXmIGJmZqXVNYhImidppqQHJLXmtN0k3VVJk7R3Tpek8yTNlfSQpD0K85kg6bH8mVBI3zPPf27Oq3ouj5mZra4RZyIHRsRuEdGSv/8AODMidgO+kb8DHAqMyZ8TgPMBJG0OfBPYB9gb+KakzXKe8/O0lXyH1H9xzMysojeaswLYJA9vCizMw0cAl0dyFzBE0tbAOGBaRDwXEc8D04BD8rhNIuLOiAjgcmB8Q5fEzKyfq3cHjAHcLCmACyLiQuBkYKqkH5KC2DvytCOApwp55+e0jtLnV0lfg6QTSGcsjBo1qpuLZGZmFfU+E9kvIvYgNVWdJGl/4ETgCxGxDfAF4OI8bbXrGVEifc3EiAsjoiUiWoYNG9bVZTAzs3bUNYhExML8dxEwmXRNYwJwbZ7k6pwG6Uxim0L2kaSmro7SR1ZJNzOzBqlbEJG0kaTBlWHgYGAW6UD/7jzZQcBjeXgKcGy+S2ss8EJEPA1MBQ6WtFm+oH4wMDWPe0nS2HxX1rHA9fVaHjMzW1M9r4lsBUzOd90OBCZGxE2SXgZ+LGkg8A/ytQrgBuAwYC7wKvBxgIh4TtK3gXvzdN+KiOfy8InApcAg4Mb8MTOzBlG6san/aGlpidbW1t6uhplZU5E0o/Coxkp+Yt3MzEpzEDEzs9IcRMzMrDQHETMzK81BxMzMSnMQMTOz0hxEzMysNAcRMzMrzUHEzMxKcxAxM7PSHETMzKw0BxEzMyvNQcTMzEpzEDEzs9IcRMzMrDQHETMzK81BxMzMSnMQMTOz0hxEzMysNAcRMzMrzUHEzMxKcxAxM7PSHETMzKw0BxEzMyvNQcTMzEpzEDEzs9IcRMzMrDQHETMzK81BxMzMSnMQMTOz0hxEzMysNAcRMzMrzUHEzMxKcxAxM7PSHETMzKw0BxEzMyutrkFE0jxJMyU9IKm1kP5ZSXMkzZb0g0L6aZLm5nHjCumH5LS5kk4tpG8r6W5Jj0n6paT16rk8Zma2ug6DiKQBkn7RzTIOjIjdIqIlz/NA4Ahgl4jYCfhhTt8ROArYCTgE+FkufwDwU+BQYEfg6DwtwFnAuRExBngeOL6bdTUzsy7oMIhExApgWA//wj8R+H5ELMtlLMrpRwCTImJZRDwBzAX2zp+5EfF4RLwGTAKOkCTgIOCanP8yYHwP1tPMzDoxsIZp5gF3SJoCvFJJjIhzasgbwM2SArggIi4EtgfeJem7wD+AL0XEvcAI4K5C3vk5DeCpNun7AFsASyLi9SrTr0bSCcAJAKNGjaqh2mZmVotagsjC/FkHGNzF+e8XEQslbQlMk/RoLnMzYCywF/ArSdsBqpI/qH62FB1Mv2ZiCl4XArS0tFSdxszMuq7TIBIRZwJI2igiXuls+jZ5F+a/iyRNJjVNzQeujYgA7pH0BjA0p29TyD6SFLxoJ/1ZYIikgflspDi9mZk1QKd3Z0naV9LDwCP5+66SflZDvo0kDa4MAwcDs4DrSNcykLQ9sB4pIEwBjpK0vqRtgTHAPcC9wJh8J9Z6pIvvU3IQuhU4Mhc5Abi+5iU3M7Nuq6U560fAONJBnoh4UNL+NeTbCpicrn8zEJgYETflQHCJpFnAa8CEHBBmS/oV8DDwOnBSvrCPpP8ApgIDgEsiYnYu46vAJEnfAe4HLq5loc3MrGfUEkSIiKdyMKhYUUOex4Fdq6S/Bny0nTzfBb5bJf0G4IZ2yti7s7qYmVl91BJEnpL0DiDyWcTnyE1bZmbWv9XyxPpngJNIt88uAHbL383MrJ+r5e6sZ4FjGlAXMzNrMrXcnbWdpN9IWixpkaTr83MdZmbWz9XSnDUR+BWwNTAcuBq4qp6VMjOz5lBLEFFEXBERr+fPL2jnyXAzM+tf2r0mImnzPHhr7n59Eil4fAT4XQPqZmZmfVxHF9ZnsHofVZ8ujAvg2/WqlJmZNYd2g0hEbNvIipiZWfPp9Bbf/FKo9wGji9PX2BW8mZmtxWp5Yv03pPd+zATeqG91zMysmdQSREZGxC51r4mZmTWdWm7xvVHSwXWviZmZNZ1azkTuInXpvg6wnHS3VkTEJnWtmZmZ9Xm1BJH/BvYFZub3fpiZmQG1NWc9BsxyADEzs7ZqORN5Gpgu6UZgWSXRt/iamVktQeSJ/Fkvf8zMzIDa3idyZiMqYmZmzaeWJ9ZvpUqvvRFxUF1qZGZmTaOW5qwvFYY3AD4EvF6f6piZWTOppTlrRpukOyTdVqf6mJlZE6mlOWvzwtd1gD2BN9WtRmZm1jRqac4qvlfkddKdWsfXs1JmZtYcamnO8ntFzMysqlrORJD0DtZ8n8jldaqTmZk1iVquiVwBvAV4AFiRkwNwEDEz6+dqORNpAXZ031lmZtZWLR0wzsJ3Y5mZWRW1nIkMBR6WdA+rd8B4eN1qZWZmTaGWIHJGvSthZmbNqZZbfP10upmZVVXLNREzM7OqHETMzKw0BxEzMyutlocN9yNdXH9znl5ARMR29a2amZn1dbWciVwMnAO8E9iL9PDhXrXMXNI8STMlPSCptc24L0kKSUPzd0k6T9JcSQ9J2qMw7QRJj+XPhEL6nnn+c3Ne1VIvMzPrGbXc4vtCRNzYjTIOjIhniwmStgHeCzxZSD4UGJM/+wDnA/vkrui/SQpeAcyQNCUins/TnADcBdwAHAJ0p65mZtYFtZyJ3CrpbEn7Stqj8ulmuecCX2H11+4eAVweyV3AEElbA+OAaRHxXA4c04BD8rhNIuLO3CXL5cD4btbLzMy6oJYzkX3y35ZCWgC1vGM9gJslBXBBRFwo6XBgQUQ82Kb1aQTwVOH7/JzWUfr8KulrkHQC6YyFUaNG1VBtMzOrRS0PGx7YjfnvFxELJW0JTJP0KPA14OAq01a7nhEl0tdMjLgQuBCgpaXFHUmamfWQdoOIpI9GxC8knVJtfESc09nMI2Jh/rtI0mTg3cC2QOUsZCRwn6S9SWcS2xSyjwQW5vQD2qRPz+kjq0xvZmYN0tE1kY3y38HtfDokaSNJgyvDpLOPeyNiy4gYHRGjSYFgj4j4GzAFODbfpTWWdEH/aWAqcLCkzSRtluczNY97SdLYfFfWscD1XV0BZmZWXrtnIhFxQf57Zsl5bwVMzmccA4GJEXFTB9PfABwGzAVeBT6ey39O0reBe/N034qI5/LwicClwCDSXVm+M8vMrIHU2bumJG0AHA/sBGxQSY+IT9S3avXR0tISra2tnU9oZmYrSZoRES1t02u5xfcK0kupxgG3ka49vNSz1TMzs2ZUSxB5a0R8HXglIi4D3ge8vb7VMjOzZlBLEFme/y6RtDOwKTC6bjUyM7OmUcvDhhfmu6K+TrqDamPgG3WtlZmZNYVaHja8KA/eBrjnXjMzW6mWruCHkJ7BGF2cPiI+V79qmZlZM6ilOesGUi+5M4E36lsdMzNrJrUEkQ0iomrXJ2Zm1r/V9JyIpE9J2lrS5pVP3WtmZmZ9Xi1nIq8BZ5N636083h74IruZWb9XSxA5hfTA4bOdTmlmZv1KLc1Zs0kdIpqZma2mljORFcADkm4FllUSfYuvmZnVEkSuyx8zM7PV1PLE+mWS1gO2z0lzImJ5R3nMzKx/qOWJ9QOAy4B5pPeabyNpQkTcXt+qmZlZX1dLc9Z/AwdHxBwASdsDVwF71rNiZmbW99Vyd9a6lQACEBF/BtatX5XMzKxZ1HIm0irpYtIbDgGOAWbUr0pmZtYsagkiJwInAZ8jXRO5HfhZPStlZmbNoZa7s5ZJugK4IiIWN6BOZmbWJNq9JqLkDEnPAo8CcyQtluS3GpqZGdDxhfWTgf2AvSJii4jYHNgH2E/SFxpSOzMz69M6CiLHAkdHxBOVhIh4HPhoHmdmZv1cR0Fk3Wo99+brIr7F18zMOgwir5UcZ2Zm/URHd2ftKunFKukCNqhTfczMrIm0G0QiYkAjK2JmZs2nlm5PzMzMqnIQMTOz0hxEzMysNAcRMzMrzUHEzMxKcxAxM7PSHETMzKw0BxEzMyvNQcTMzEqraxCRNE/STEkPSGrNaWdLelTSQ5ImSxpSmP40SXMlzZE0rpB+SE6bK+nUQvq2ku6W9JikX0par57LY2Zmq2vEmciBEbFbRLTk79OAnSNiF+DPwGkAknYEjgJ2Ag4BfiZpgKQBwE+BQ4EdgaPztABnAedGxBjgeeD4BiyPmZllDW/OioibI+L1/PUuYGQePgKYFBHL8jtM5gJ758/ciHg8Il4DJgFHSBJwEHBNzn8ZML5Ry2FmZvUPIgHcLGmGpBOqjP8EcGMeHgE8VRg3P6e1l74FsKQQkCrpa5B0gqRWSa2LF/s18WZmPaXeQWS/iNiD1BR1kqT9KyMkfQ14HbiyklQlf5RIXzMx4sKIaImIlmHDhnWl/mZm1oG6BpGIWJj/LgImk5qmkDQBeD9wTERUDvzzgW0K2UcCCztIfxYYImlgm3QzM2uQugURSRtJGlwZBg4GZkk6BPgqcHhEvFrIMgU4StL6krYFxgD3APcCY/KdWOuRLr5PycHnVuDInH8CcH29lsfMzNbU0ZsNu2srYHK6/s1AYGJE3CRpLrA+MC2PuysiPhMRsyX9CniY1Mx1UkSsAJD0H8BUYABwSUTMzmV8FZgk6TvA/cDFdVweMzNrQ6tak/qHlpaWaG1t7e1qmJk1FUkzCo9qrOQn1s3MrDQHETMzK81BxMzMSnMQMTOz0hxEzMysNAcRMzMrzUHEzMxKcxAxM7PSHETMzKw0BxEzMyvNQcTMzEpzEDEzs9Lq2YuvmZl103X3L+DsqXNYuGQpw4cM4svjdmD87lVf4torHETMzPqo6+5fwGnXzmTp8hUALFiylNOunQnQZwKJm7PMzPqos6fOWRlAKpYuX8HZU+f0Uo3W5CBiZtZHLVyytEvpvcFBxMysjxo+ZFCX0nuDg4iZWR/15XE7MGjdAaulDVp3AF8et0Mv1WhNvrBuZtZHVS6e++4sMzMrZfzuI/pU0GjLzVlmZlaag4iZmZXmIGJmZqU5iJiZWWkOImZmVprvzjIzq6O+3oFidzmImNlarbsH8e7kb4YOFLvLQcTM+rTePIh3N39HHSg6iJh1QW/+Guzt/M1c997O39sH8e7mb4YOFLvLQaRJ9OcDQTPnb+a694X8vX0Q727+4UMGsaDKtH2pA8Xu8t1ZTaDyj7hgyVKCVf+I192/oCnyd/edCM2cv5nr3hfy98RBvCvpPZ2/GTpQ7C4HkSbQ2//IvX0gaOb8zVz3vpC/tw/i3c0/fvcR/Ne/vJ0RQwYhYMSQQfzXv7x9rbkeAm7Oagq9/Y/c26f0zZy/meveF/J/edwOqzWHQdcP4lC+F9ye6EW3r3eg2F0+E2kCvX1K3uy/BnszfzPXvS/k74lf8uN3H8Edpx7EE99/H3ecelCXD+jdzb+285lIE+jur7Hezt/bvwZ7M38z170v5K/MwwfuvksR0dt1aKiWlpZobW3t7Wp0WTPfnWVmzU/SjIhoWSO9nkFE0jzgJWAF8HpEtEjaHPglMBqYB3w4Ip6XJODHwGHAq8BxEXFfns8E4PQ82+9ExGU5fU/gUmAQcAPw+ehkgcoGER+Ezaw/ay+INOKayIERsVuh8FOBWyJiDHBL/g5wKDAmf04AzgfIQeebwD7A3sA3JW2W85yfp63kO6QeC9Dbt8iamfVVvXFh/Qjgsjx8GTC+kH55JHcBQyRtDYwDpkXEcxHxPDANOCSP2yQi7sxnH5cX5tWjevsWWTOzvqreQSSAmyXNkHRCTtsqIp4GyH+3zOkjgKcKeefntI7S51dJX4OkEyS1SmpdvHhxlxeit2+RNTPrq+odRPaLiD1ITVUnSdq/g2lVJS1KpK+ZGHFhRLRERMuwYcM6q/MaevsWWTOzvqquQSQiFua/i4DJpGsaz+SmKPLfRXny+cA2hewjgYWdpI+skt7jevteeTOzvqpuQUTSRpIGV4aBg4FZwBRgQp5sAnB9Hp4CHKtkLPBCbu6aChwsabN8Qf1gYGoe95KksfnOrmML8+pR3X3gqT90fWBm/VPdbvGVtB3p7APSQ40TI+K7krYAfgWMAp4E/jUinsuB4CekO6xeBT4eEa15Xp8A/jPP67sR8fOc3sKqW3xvBD5br1t8zcz6s155TqQvchAxM+u63nxOxMzM1lIOImZmVpqDiJmZleYgYmZmpfW7C+uSXgJ6s7+RocCzLr9flt+fl93lN3/5b46INZ7W7o/vE5lT7Q6DRpHU6vL7Z/n9edld/tpbvpuzzMysNAcRMzMrrT8GkQtdvsvvh2W7fJdfl/L73YV1MzPrOf3xTMTMzHqIg4iZmZXWb4KIpEMkzZE0V9KpnefodnmXSFokaVYhbXNJ0yQ9lv9u1tE8uln+NpJulfSIpNmSPt/IOkjaQNI9kh7M5Z+Z07eVdHcu/5eS1qtH+YV6DJB0v6TfNrp8SfMkzZT0gKRKj9SN3AeGSLpG0qN5P9i3gdt/h7zclc+Lkk5u8PJ/Ie97syRdlffJRm7/z+eyZ0s6OafVbfm7csxJb9zQefl4+JCkPcqW2y+CiKQBwE9Jb1jcETha0o51LvZSUrf2RacCt0TEGOCW/L1eXge+GBFvA8aS3iy5YwPrsAw4KCJ2BXYDDlF6T8xZwLm5/OeB4+tUfsXngUcK3xtd/oERsVvh/vxG7gM/Bm6KiH8CdiWth4aUHxFz8nLvBuxJer3D5EaVL2kE8DmgJSJ2BgYAR9Gg7S9pZ+BTpBfx7Qq8X9IY6rv8l1L7MedQYEz+nACcX7rUiFjrP8C+pBdZVb6fBpzWgHJHA7MK3+cAW+fhrUkPPjZqHVwPvLc36gBsCNwH7EN6YnZgte1Sh3JH5n+cg4Dfkl6p3Mjy5wFD26Q1ZP0DmwBPkG+e6c19kPQiuTsavPwjgKeAzUkPVf8WGNeo7Q/8K3BR4fvXga/Ue/lrPeYAFwBHV5uuq59+cSbCqh2qYn5Oa7StIr2Rkfx3y0YUKmk0sDtwdyPrkJuSHiC9Anka8BdgSUS8niep93b4Eekf9438fYsGlx/AzZJmSDohpzVq/W8HLAZ+npvzLlJ6w2hv7INHAVfl4YaUHxELgB+SXnz3NPACMIPGbf9ZwP6StpC0IXAY6TXfjV7/7ZXXY8fE/hJEVCWtX9zbLGlj4NfAyRHxYiPLjogVkZozRpJO699WbbJ6lC3p/cCiiJhRTG5U+dl+EbEHqengJEn717GstgYCewDnR8TuwCvUt+msqnzN4XDg6gaXuxlwBLAtMBzYiLQd2qrL9o+IR0hNZ9OAm4AHSU3MfUWP/S/0lyAyn/QroGIksLAX6vGMpK0B8t9F9SxM0rqkAHJlRFzbG3UAiIglwHTStZkhkip9ttVzO+wHHC5pHjCJ1KT1owaWT0QszH8Xka4H7E3j1v98YH5E3J2/X0MKKo3e/ocC90XEM/l7o8p/D/BERCyOiOXAtcA7aOz2vzgi9oiI/YHngMdo/Ppvr7weOyb2lyByLzAm35mxHun0ekov1GMKMCEPTyBdp6gLSQIuBh6JiHMaXQdJwyQNycODSP/UjwC3AkfWu/yIOC0iRkbEaNL2/n1EHNOo8iVtJGlwZZh0XWAWDVr/EfE34ClJO+SkfwYeblT5BUezqimLBpb/JDBW0ob5f6Gy/A3Z/gCStsx/RwH/QloPjV7/7ZU3BTg236U1Fnih0uzVZfW4qNQXP6Q2yT+T2uW/1oDyriK1xS4nRf3jSW3yt4stru8AAAmrSURBVJB+kdwCbF7H8t9JOj19CHggfw5rVB2AXYD7c/mzgG/k9O2Ae4C5pCaO9RuwLQ4AftvI8nM5D+bP7Mo+1+B9YDegNW+D64DNGlz+hsDfgU0LaY0s/0zg0bz/XQGs38j9D/gDKXA9CPxzvZe/K8ccUnPWT/PxcCbpLrZS5brbEzMzK62/NGeZmVkdOIiYmVlpDiJmZlaag4iZmZXmIGJmZqU5iPRxklbkXlBnSbo6d6HQ0fQ3VJ7PKFHWRT3dMaVST7ZDe3KebebfIum8es2/KyRNl9TS+ZS9R9Lh6mYv1u0tp6R35R5rH8jPBq01JL3cTvogSbflLn52yF3cPChp3zx+oKT/K/7fSpqUO2NcKziI9H1LI/WGujPwGvCZjiaOiMMiPSG+Un6gqNNtHRGfjIiHu1fdxoqI1oj4XG/XI/cUXY/5Dux8qtpFxJSI+H6dyjkG+GHeX5d2NnFn66xe67SHfQK4NiJWAJ8mdS1zJPClPP5E4IqIeLWQ53xSn25rBQeR5vIH4K0Akq7Lv3pmFzr3W/nLX9JopXdI/IzUg+7HJJ2Tp/m8pMfz8Fsk/TEPT8+/7AdIujSf/cyU9IXCtDflcv8g6Z/aVjB3OHdz7vTvAgp99Eg6Jc9zlla9X2G00vsuLsrpV0p6j6Q7lN6BsHeebm9Jf8rz/VPlSWxJB2jVu0LOUHqnwnRJj0uqGlwkHZ2Xa5aks3LaiZJ+UJjmOEn/k4c/qvRulAckXVA5uEl6WdK3JN1N6hG2WMb5klpVeJdKYfucled3j6S3VqnfGZIulHQzcHneHmdLulfp3Q+fztNJ0k8kPSzpd0pnoUcW94M83CJpemG5fpKHL5V0jqRbgbOUnrK/JJdzv6Qj8nSD8q/nhyT9EljjLEPSJ4EPA9/I21C5zpV96COF7XWrpImkh9zazme1dSppT6Vf+jMkTdWqLjymSzpX0u1K+/lekq7N+8x3OtnnzpL0723W9xclbSzpFkn35TofUW3/aeMYVj0Fvjyvmw2B5UotAh8ALm+T5w/Ae9TDPxB6Tb2e1vSnx55CfTn/HUjaWU/M3ytPng4iPZG7Rf4+DxhK6hL6DWBsTn8TcG8evobUFcwIUlcI/5XTpwMtpPc/TCvUYUj+ewswJg/vQ+pKpG19z2PV0+nvIz01PzTPcyapI7yNSU9x757r+TrwdtKPmhnAJaTgcwRwXZ7XJqzqwvs9wK/z8AGsehr9DOBPpCeTh5Kell63Tf2Gk7rEGJbX6e+B8fn73MJ0N5Ke+n8b8JvKfICfAcfm4QA+XMgznfzkb2H7DMjpuxS2T+Xp9WMrdW9TxzPyehiUv58AnJ6H1yc9hb4tqSuNabmM4cAS4MjifpCHW4Dpefg44Cd5+FJSF+kD8vfvAR+tbHNSDw8bAacAl+T0XfL2WuMJ5zy/SvkfKtRtq7zOt87b6xVg23b295XrFFg3b89h+ftHCvWYDpyVhz9P6vdp67x+5pOe1G5vn9sduK1Q5sPAKNL+sElOG0p6qr3yQPbLVeq6HvC3wvdRuV535vV0DvDudpZzGrBnbx9feuKzdkTCtdsgpe7UIf2CuTgPf07SB/PwNqSXy/y9Td6/RsRdkPpSyr+0BufpJwL7A+8idU5X9DiwXf4l/jtSd+Ybkzqwu1paeXKxfpX67k86uBERv5P0fE5/JzA5Il4BkHRtLnsKqaO8mTl9NuklOiFpJinIAGwKXKbUlhykA0w1v4uIZcAySYtIB7D5hfF7kQ6oi3N5VwL7R8R1+exlLKmLiB2AO4CTSAeje/NyD2JVJ3YrSB1cVvNhpTPEgaSD246k7kdgVV9SVwHntpN/SqxqEjoY2KVylpHXxRjSur4qUlPKQkm/b2deHbk656+Uc7ikSlPMBqQD4/6kHwdExEOSHlpzNmt4Z6Fuz0i6jbTuXwTuiYgn2slXXKc7ADsD0/K6H0Dq1qOi0v/dTGB25L6flM6yt6GdfS4izpO0paThpB8Pz0fEk0odln5PqbflN0g/srYC/tZOXYeSAjd53TxJCpLkM8zhwKOSriAFnK9HxJ/z5Ivy+GIv003JQaTvWxqpO/WVJB1A+jW+b0S8mpsqNqiS95U23+8EPk56Ac0fSO25+wJfLE4UEc9L2pX0Ep+TSM0UJ5PexbBaXdpRrS+dal1PVywrDL9R+P4Gq/bRbwO3RsQHld6PMr2Gea1gzX28o3r8krSsj5IOPqF09LosIk6rMv0/CgfgVQVI25LaxPfK6/JSVt8+0c5wUXHbCfhsRExtU85hHeR/nVXN1dX2jfbK+VBEzGlTTkf1bE9H67ntfllUXKciBYd925m2uJ+03YcGdlKHa0jXLt5E6uUZUtPUMNIZwnKlHqA7WndLOxj/XeB00tsVrySdGX4zl0HO1+l1o2bgayLNaVPSr6dXla5LjK0x3+2kg9vtpM4RDwSWRcQLxYlyW/o6EfFr0hvZ9oj0LpInJP1rnkY50FQr45g8zaGkTv8q6eOVelXdCPggKZDValNgQR4+rgv52robeLfSdaMBpF5mb8vjriU1bR1NCiiQmvCO1KoeWTeX9OZOytiEdKB8QdJWrPkei48U/t5ZQ52nAifmX8pI2j6vw9uBo5SumWxN2p4V80hnUJCalmoxFfhsDpxI2j2nF7fpzqSmms7cDnwk120Y6WzmnhrrUTEHGKZVdzqtK2mnLuTvaJ+bROrd+UhSQIG0jy3KAeRAoMPtHBHPAwMkrRZIJL0bWBARj5Guj7xB+kFTvLNye1LzWtPzmUhzugn4TG5WmAPcVWO+P5BO82+PiBWSniL96m5rBOmNeJUfGZVf4ccA50s6ndScNInUQ2nRmcBVku4jHZyfBIiI+/Iv8sqB5KKIuD+fVdTiB6TmrFNI1zFKiYinJZ1G6hJcwA0RcX0e97ykh4EdI+KenPZwXt6b8/pYTjo7+2sHZTwo6X7SQeJxUrNY0fr5wvE6pIDVmYtIzXr35QP8YlKwm0x6T8pM0vWL2wp5zgQulvSfpMBZi2+T3rnyUC5nHvB+0t1EP8/72wPUFgwmk85yHySdxXwlN6mucTNGeyLitdyEd56kTUnHqx9R48G3vX0uj5udm3YXxKou0K8EfiOplbSc1f432rqZ1Gz2f7DyFQynk85oAS7M8x1IulOL/MNiaZTter2PcS++Zg2Um0haIuLZOsz7UtKF+ms6m9Z6Rj5bOyUiPtaFPF8AXoyIizuduAm4OcvMrKR8ZnOruvZMyxLgsjpVqeF8JmJmZqX5TMTMzEpzEDEzs9IcRMzMrDQHETMzK81BxMzMSvv/kT/dksaaoCMAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "def test_overlap(pos1, pos2, cutoff):\n",
    "    '''\n",
    "    Find out whether two defined regions overlap or not.\n",
    "    Overlap is defined True when at least one of the domains \n",
    "    overlap with the other above a certain fraction of the whole.\n",
    "    '''\n",
    "    pos1 = [int(s) for s in pos1]\n",
    "    pos2 = [int(s) for s in pos2]\n",
    "    both_pairs = sorted([pos1, pos2])\n",
    "    \n",
    "    # simplest case, they do not overlap at all\n",
    "    if both_pairs[0][1] < both_pairs[1][0]:\n",
    "        return False\n",
    "    \n",
    "    else:\n",
    "        # get overlap part\n",
    "        overlap_region = both_pairs[0][1] - both_pairs[1][0]\n",
    "        \n",
    "        # fraction of first domain\n",
    "        frac_first = overlap_region / (both_pairs[0][1] - both_pairs[0][0])\n",
    "                          \n",
    "        # fraction of second domain\n",
    "        frac_second = overlap_region / (both_pairs[1][1] - both_pairs[1][0])\n",
    "        \n",
    "        if max(frac_first, frac_second) > cutoff:\n",
    "            #print(both_pairs, True, max(frac_first, frac_second))\n",
    "            return True\n",
    "        else:\n",
    "            #print(both_pairs, False, max(frac_first, frac_second))\n",
    "            return False\n",
    "    \n",
    "\n",
    "    \n",
    "    \n",
    "def remove_overlaps(data, cutoff):\n",
    "    '''\n",
    "    When domains overlap (as defined by the cutoff level), \n",
    "    keep only the domain with the best e-value score.\n",
    "    '''\n",
    "    overlaps = [True]\n",
    "    while any(overlaps):\n",
    "        \n",
    "        start_end_pairs = list(zip(data.gene_match_from.values, data.gene_match_to.values))\n",
    "        e_vals = data.match_evalue.values\n",
    "        e_vals = e_vals.astype(np.float) \n",
    "\n",
    "        \n",
    "        for i in range(len(start_end_pairs)):\n",
    "            overlaps = []\n",
    "            for j in range(len(start_end_pairs[i+1:])):\n",
    "\n",
    "                overlaps.append(test_overlap(start_end_pairs[i], start_end_pairs[i+1+j], cutoff))\n",
    "\n",
    "            # prepend true or false for the first sequence, depending on whether there was an overlap\n",
    "            if any(overlaps):\n",
    "                overlaps = [False for s in range(i)] + [True] + overlaps\n",
    "\n",
    "                # from the ones with an overlap, what is the position of the one with the lowest e-value\n",
    "                overlap_e_vals = e_vals[overlaps]\n",
    "\n",
    "                idx = overlap_e_vals.argmin()\n",
    "                #print(overlaps)\n",
    "                #print(overlap_e_vals)\n",
    "                #print(i+idx)\n",
    "\n",
    "\n",
    "                # flip the one that said true but had the smallest e-value\n",
    "                overlaps[i+idx] = False\n",
    "\n",
    "                # invert the list to flip True vs False\n",
    "                overlaps = [not i for i in overlaps]\n",
    "\n",
    "                # filter the data to keep all the ones that originally said False, but now say true\n",
    "                data = data[overlaps]\n",
    "                #print(data)\n",
    "                break\n",
    "                \n",
    "    return data\n",
    "\n",
    "\n",
    "    \n",
    "# test a range of cutoff values to figure out which one is best\n",
    "cutoff_vals = [s/100 for s in range(5, 105, 5)]\n",
    "motifs = []\n",
    "for cutoff in cutoff_vals:\n",
    "    non_redundant_data = all_genome_data.groupby('gene').apply(lambda x: remove_overlaps(x, cutoff=cutoff))\n",
    "    motifs.append(non_redundant_data.gene.count())\n",
    "\n",
    "\n",
    "print('Unfiltered domain number: %s' % all_genome_data.gene.count())\n",
    "\n",
    "\n",
    "\n",
    "# plot the data\n",
    "plt.scatter([s * 100 for s in cutoff_vals], motifs)\n",
    "plt.xlabel('Pairwise domain overlap required for removal (%)')\n",
    "plt.ylabel('Domain number')\n",
    "plt.xticks(np.arange(0, 100+10, 10))\n",
    "plt.title('Domain overlap sensitivity analysis')\n",
    "\n",
    "\n",
    "plt.savefig('./figures/domain_sensitivity_analysis.png', dpi=300, facecolor='w', edgecolor='w',\n",
    "        orientation='portrait', papertype=None, format=None,\n",
    "        transparent=False, bbox_inches=None, pad_inches=0.1, metadata=None)\n",
    "\n",
    "plt.savefig('./figures/domain_sensitivity_analysis.pdf', facecolor='w', edgecolor='w',\n",
    "        orientation='portrait', papertype=None, format=None,\n",
    "        transparent=False, bbox_inches=None, pad_inches=0.1, metadata=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Now remove redundant domains using the most appropriate cutoff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter out overlapping domains using pre-determined cutoff value\n",
    "non_redundant_data = all_genome_data.groupby('gene').apply(lambda x: remove_overlaps(x, cutoff=0.2))\n",
    "\n",
    "# write file to disk\n",
    "non_redundant_data.to_csv(join(FINAL, '332_yeast_genomes_enzyme_info_version_3.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Output a table with the number of domains per organism"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>family</th>\n",
       "      <th>organism</th>\n",
       "      <th>AA1</th>\n",
       "      <th>AA11</th>\n",
       "      <th>AA2</th>\n",
       "      <th>AA3</th>\n",
       "      <th>AA4</th>\n",
       "      <th>AA5</th>\n",
       "      <th>AA6</th>\n",
       "      <th>AA7</th>\n",
       "      <th>AA9</th>\n",
       "      <th>...</th>\n",
       "      <th>GT83</th>\n",
       "      <th>GT90</th>\n",
       "      <th>GT91</th>\n",
       "      <th>PL1</th>\n",
       "      <th>PL26</th>\n",
       "      <th>PL3</th>\n",
       "      <th>PL33</th>\n",
       "      <th>PL35</th>\n",
       "      <th>PL4</th>\n",
       "      <th>PL7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>aciculoconidium_aculeatum</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>alloascoidea_hylecoeti</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ambrosiozyma_ambrosiae</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ambrosiozyma_kashinagacola</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ambrosiozyma_maleeae</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 147 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "family                    organism  AA1  AA11  AA2  AA3  AA4  AA5  AA6  AA7  \\\n",
       "0        aciculoconidium_aculeatum  5.0   0.0  1.0  1.0  4.0  0.0  7.0  0.0   \n",
       "1           alloascoidea_hylecoeti  3.0   0.0  4.0  0.0  4.0  0.0  4.0  1.0   \n",
       "2           ambrosiozyma_ambrosiae  6.0   0.0  2.0  0.0  2.0  0.0  5.0  1.0   \n",
       "3       ambrosiozyma_kashinagacola  5.0   0.0  2.0  0.0  4.0  0.0  3.0  1.0   \n",
       "4             ambrosiozyma_maleeae  4.0   0.0  2.0  0.0  2.0  0.0  4.0  1.0   \n",
       "\n",
       "family  AA9  ...  GT83  GT90  GT91  PL1  PL26  PL3  PL33  PL35  PL4  PL7  \n",
       "0       0.0  ...   0.0   0.0   1.0  0.0   0.0  0.0   0.0   0.0  0.0  9.0  \n",
       "1       2.0  ...   0.0   0.0   0.0  4.0   0.0  0.0   0.0   0.0  1.0  0.0  \n",
       "2       0.0  ...   0.0   0.0   1.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "3       0.0  ...   0.0   0.0   3.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "4       0.0  ...   0.0   0.0   3.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "\n",
       "[5 rows x 147 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>family</th>\n",
       "      <th>organism</th>\n",
       "      <th>AA1</th>\n",
       "      <th>AA11</th>\n",
       "      <th>AA2</th>\n",
       "      <th>AA3</th>\n",
       "      <th>AA4</th>\n",
       "      <th>AA5</th>\n",
       "      <th>AA6</th>\n",
       "      <th>AA7</th>\n",
       "      <th>AA9</th>\n",
       "      <th>...</th>\n",
       "      <th>GT83</th>\n",
       "      <th>GT90</th>\n",
       "      <th>GT91</th>\n",
       "      <th>PL1</th>\n",
       "      <th>PL26</th>\n",
       "      <th>PL3</th>\n",
       "      <th>PL33</th>\n",
       "      <th>PL35</th>\n",
       "      <th>PL4</th>\n",
       "      <th>PL7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>327</td>\n",
       "      <td>zygosaccharomyces_bisporus</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>328</td>\n",
       "      <td>zygosaccharomyces_kombuchaensis</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>329</td>\n",
       "      <td>zygosaccharomyces_rouxii</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>330</td>\n",
       "      <td>zygotorulaspora_florentina</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>331</td>\n",
       "      <td>zygotorulaspora_mrakii</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 147 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "family                         organism  AA1  AA11  AA2  AA3  AA4  AA5  AA6  \\\n",
       "327          zygosaccharomyces_bisporus  5.0   0.0  1.0  0.0  3.0  0.0  3.0   \n",
       "328     zygosaccharomyces_kombuchaensis  4.0   0.0  2.0  1.0  2.0  0.0  3.0   \n",
       "329            zygosaccharomyces_rouxii  4.0   0.0  1.0  2.0  3.0  0.0  3.0   \n",
       "330          zygotorulaspora_florentina  5.0   0.0  2.0  0.0  1.0  0.0  2.0   \n",
       "331              zygotorulaspora_mrakii  4.0   0.0  2.0  0.0  2.0  0.0  3.0   \n",
       "\n",
       "family  AA7  AA9  ...  GT83  GT90  GT91  PL1  PL26  PL3  PL33  PL35  PL4  PL7  \n",
       "327     1.0  0.0  ...   0.0   0.0   1.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "328     1.0  0.0  ...   0.0   0.0   1.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "329     1.0  0.0  ...   0.0   0.0   0.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "330     2.0  0.0  ...   0.0   0.0   0.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "331     2.0  0.0  ...   0.0   0.0   0.0  0.0   0.0  0.0   0.0   0.0  0.0  0.0  \n",
       "\n",
       "[5 rows x 147 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>family</th>\n",
       "      <th>AA1</th>\n",
       "      <th>AA11</th>\n",
       "      <th>AA2</th>\n",
       "      <th>AA3</th>\n",
       "      <th>AA4</th>\n",
       "      <th>AA5</th>\n",
       "      <th>AA6</th>\n",
       "      <th>AA7</th>\n",
       "      <th>AA9</th>\n",
       "      <th>CBM1</th>\n",
       "      <th>...</th>\n",
       "      <th>GT83</th>\n",
       "      <th>GT90</th>\n",
       "      <th>GT91</th>\n",
       "      <th>PL1</th>\n",
       "      <th>PL26</th>\n",
       "      <th>PL3</th>\n",
       "      <th>PL33</th>\n",
       "      <th>PL35</th>\n",
       "      <th>PL4</th>\n",
       "      <th>PL7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>count</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "      <td>332.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>mean</td>\n",
       "      <td>3.358434</td>\n",
       "      <td>0.048193</td>\n",
       "      <td>1.472892</td>\n",
       "      <td>0.740964</td>\n",
       "      <td>2.686747</td>\n",
       "      <td>0.030120</td>\n",
       "      <td>2.954819</td>\n",
       "      <td>1.096386</td>\n",
       "      <td>0.099398</td>\n",
       "      <td>0.846386</td>\n",
       "      <td>...</td>\n",
       "      <td>0.006024</td>\n",
       "      <td>0.141566</td>\n",
       "      <td>2.066265</td>\n",
       "      <td>0.075301</td>\n",
       "      <td>0.033133</td>\n",
       "      <td>0.015060</td>\n",
       "      <td>0.009036</td>\n",
       "      <td>0.009036</td>\n",
       "      <td>0.072289</td>\n",
       "      <td>0.030120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>std</td>\n",
       "      <td>1.397393</td>\n",
       "      <td>0.228147</td>\n",
       "      <td>0.774901</td>\n",
       "      <td>1.001091</td>\n",
       "      <td>1.304676</td>\n",
       "      <td>0.217781</td>\n",
       "      <td>1.155994</td>\n",
       "      <td>0.751465</td>\n",
       "      <td>0.586144</td>\n",
       "      <td>2.228059</td>\n",
       "      <td>...</td>\n",
       "      <td>0.077498</td>\n",
       "      <td>1.025868</td>\n",
       "      <td>2.954101</td>\n",
       "      <td>0.619710</td>\n",
       "      <td>0.195381</td>\n",
       "      <td>0.121976</td>\n",
       "      <td>0.122572</td>\n",
       "      <td>0.094771</td>\n",
       "      <td>0.281692</td>\n",
       "      <td>0.496814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>min</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25%</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50%</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>75%</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>max</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>8.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>16.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>19.000000</td>\n",
       "      <td>9.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>9.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 146 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "family         AA1        AA11         AA2         AA3         AA4  \\\n",
       "count   332.000000  332.000000  332.000000  332.000000  332.000000   \n",
       "mean      3.358434    0.048193    1.472892    0.740964    2.686747   \n",
       "std       1.397393    0.228147    0.774901    1.001091    1.304676   \n",
       "min       0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "25%       2.000000    0.000000    1.000000    0.000000    2.000000   \n",
       "50%       3.000000    0.000000    2.000000    0.000000    3.000000   \n",
       "75%       4.000000    0.000000    2.000000    1.000000    4.000000   \n",
       "max      10.000000    2.000000    5.000000    6.000000    7.000000   \n",
       "\n",
       "family         AA5         AA6         AA7         AA9        CBM1  ...  \\\n",
       "count   332.000000  332.000000  332.000000  332.000000  332.000000  ...   \n",
       "mean      0.030120    2.954819    1.096386    0.099398    0.846386  ...   \n",
       "std       0.217781    1.155994    0.751465    0.586144    2.228059  ...   \n",
       "min       0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "25%       0.000000    2.000000    1.000000    0.000000    0.000000  ...   \n",
       "50%       0.000000    3.000000    1.000000    0.000000    0.000000  ...   \n",
       "75%       0.000000    3.000000    1.000000    0.000000    0.000000  ...   \n",
       "max       3.000000    8.000000    6.000000    7.000000   16.000000  ...   \n",
       "\n",
       "family        GT83        GT90        GT91         PL1        PL26  \\\n",
       "count   332.000000  332.000000  332.000000  332.000000  332.000000   \n",
       "mean      0.006024    0.141566    2.066265    0.075301    0.033133   \n",
       "std       0.077498    1.025868    2.954101    0.619710    0.195381   \n",
       "min       0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "25%       0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "50%       0.000000    0.000000    1.000000    0.000000    0.000000   \n",
       "75%       0.000000    0.000000    3.000000    0.000000    0.000000   \n",
       "max       1.000000   12.000000   19.000000    9.000000    2.000000   \n",
       "\n",
       "family         PL3        PL33        PL35         PL4         PL7  \n",
       "count   332.000000  332.000000  332.000000  332.000000  332.000000  \n",
       "mean      0.015060    0.009036    0.009036    0.072289    0.030120  \n",
       "std       0.121976    0.122572    0.094771    0.281692    0.496814  \n",
       "min       0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "25%       0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "50%       0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "75%       0.000000    0.000000    0.000000    0.000000    0.000000  \n",
       "max       1.000000    2.000000    1.000000    2.000000    9.000000  \n",
       "\n",
       "[8 rows x 146 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# keep only relevant columns and count the enzyme class occurances\n",
    "heatmap_data = non_redundant_data[['organism', 'enzyme', 'family']]\n",
    "heatmap_data_count = heatmap_data.groupby(['organism', 'family']).count().reset_index()\n",
    "\n",
    "# rename columns and cast to wide format\n",
    "heatmap_data_count = heatmap_data_count.rename(columns={'organism': 'organism', 'family':'family', 'enzyme':'count'})\n",
    "heatmap_data_count = heatmap_data_count.pivot_table(index=['organism'], columns='family', values='count').reset_index()\n",
    "\n",
    "# replace NaN with 0\n",
    "heatmap_data_count = heatmap_data_count.fillna(0)\n",
    "\n",
    "\n",
    "display(heatmap_data_count.head())\n",
    "display(heatmap_data_count.tail())\n",
    "display(heatmap_data_count.describe())\n",
    "\n",
    "\n",
    "heatmap_data_count.to_csv(join(FINAL, '332_yeast_genomes_heatmap_data.tsv'), sep='\\t', index=False)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
