ETE toolkit 3.1.2
CD-HIT 4.8.1
unzip dbcan file
update info for tempo_in_genome folders

## Methods
A bioinformatic analysis was carried out with the goal of identifying carbohydrate-active enzymes in 332 published yeast genomes.

### Computational environment
The analysis was carried out in an Anaconda environment (version 2018.12) using Python (version 3.7.1) on a Unix system. Imported packages include Pandas (version 0.25.1), re (version 2.2.1) and BioPython (version 1.72). Other dependencies include HMMER, SignalP, and Tome, as outlined in detail below. The complete code used in this analysis can be found in the accompanying files "analyze_genomes.ipynb" (Jupyter notebook) and "analyze_genomes.html" (html file).

### Obtaining sequence data
Fasta files of the yeast genomes had been generated and published by another group (doi: 10.1016/j.cell.2018.10.023). The dataset (doi: 10.6084/m9.figshare.5854692) was downloaded from Figshare in November 2019. The sequences contained in the "pep" folder of this dataset was used for the analysis.

### Running dbCAN HMMs
To identify carbohydrate-active enzymes in these fasta files we first downloaded Hidden Markov Models (HMMs) from dbCAN (http://bcb.unl.edu/dbCAN2/, version 8 from the 8th of August 2019). Each sequence in the fasta files were matched against these models using HMMER (http://hmmer.org/, version 3.2.1) with the -E flag set to filter hits with e-values below 10^-15 as well as with the --domtblout flag to obtain an easily parsable output file. The output files were parsed in using Python and hits covering less than 35% of the HMM model were discarded.

### Running SignalP
For each of the enzymes in the fasta files we predicted potential signal peptides using SignalP (http://www.cbs.dtu.dk/services/SignalP/, version 5.0b). In these runs we set the -org flag to "euk" (for eukaryote), and the -format flag to "short" to obtain easily parsable output files. The output files were parsed using Python scripts.

### Predicting growth temperature
For each of the yeast genomes the growth temperature was predicted using Tome (https://github.com/EngqvistLab/Tome, version 1.1) and the output directed to files that could be subsequently parsed using Python scripts.

### Output file description
The files generated by HMMER, SignalP and Tome were parsed and combined into the "332_yeast_genomes_enzyme_info_version_3.tsv" tab-separated output file (found in the `/data/final/` folder). Each row in the data file indicated one gene with a single corresponding hmm hit at a specific position in the gene. A gene can (and often does) occur multiple times with different hmm model hits or hits with the same hmm model but at different positions inside the gene. Below follows a description of the data contained in each column of the output file.

column: organism
description: the organism name
value type: text

column: gene
description: the gene name as given inside fasta files
value type: text

column: hmm_model
description: the hmm model from signalp that gave the hit
value type: text

column: hmm_model_len
description: length of the hmm model, specified in the hmmer output file (there in the "qlen" column)
value type: integer

column: hmm_match_from
description: where in the hmm model the match with the gene starts,  specified in the hmmer output file (there in the "hmm coord from" column)
value type: integer

column: hmm_match_to
description: where in the hmm model the match with the gene ends,  specified in the hmmer output file (there in the "hmm coord to" column)
value type: integer

column: hmm_match_coverage
description: how much of hmm model actualy matched to the gene from 0.35 to 1.0, computed as ("hmm_match_to" - "hmm_match_from")/"hmm_model_len"
value type: float

column: match_evalue
description: the e-value of the hmm model hit, specified in the hmmer output file (there in the "Evalue" column)
value type: float, scientific notation

column: gene_match_from
description: where in the gene the hmm model match starts, specified in the hmmer output file (there in the "ali coord from" column)
value type: integer

column: gene_match_to
description: where in the gene the hmm model match ends, specified in the hmmer output file (there in the "ali coord to" column)
value type: integer

column: enzyme
description: the full enzyme name, parsed from the hmm model name by excluding the ".hmm" file extension
value type: text

column: family
description: the enzyme name, excluding subfamily designations, parsed form the "enzyme" column
value type: text

column: enzyme_type
description: which main class of enzyme it is, GH, CBM, CE, etc., parsed from the "family" column
value type: text

column: signal_peptide
description: whether signal peptide is predicted (SP(Sec/SPI)) or not (OTHER), specified in the signalp output file (there in the "Prediction" column)
value type: text

column: signal_peptide_prob
description: probability that a signal peptide is present, specified in the signalp output file (there in the "SP(SEC/SPI)" column)
value type: float

column: sp_cut_pos
description: the position in the protein sequence where the signal peptide is predicted to be cleaved, specified in the signalp output file (there in the "CS Position" column)
value type: text

column: sp_cut_seq
description: the sequence at which the signal peptide is predicted to be cleaved, specified in the signalp output file (there in the "CS Position" column)
value type: text

column: sp_cut_prob
description: the probability of the cut-site prediction, specified in the signalp output file (there in the "CS Position" column)
value type: float

column: genes_in_fasta
description: the number of genes present in the organisms fasta file
value type: integer

column: pred_growth_temp(c)
description: the organism growth temperature in centigrade (Celsius) as predicted by Tome from the fasta files
value type: integer


### Data visualization
To be completed once analysis is done..
